{"metadata": {"kernelspec": {"display_name": "Python 3 (ipykernel)", "language": "python", "name": "python3"}, "language_info": {"name": "python", "version": "3.11.5", "mimetype": "text/x-python", "codemirror_mode": {"name": "ipython", "version": 3}, "pygments_lexer": "ipython3", "nbconvert_exporter": "python", "file_extension": ".py"}}, "nbformat_minor": 5, "nbformat": 4, "cells": [{"cell_type": "markdown", "source": "# Initialize Notebook", "metadata": {"tags": []}, "id": "69755b57-9eee-4b3b-8dd4-6140486a4182"}, {"cell_type": "markdown", "source": "## Import packages", "metadata": {"tags": []}, "id": "b11a388c-5ca1-4097-91a5-29e8dc6bd559"}, {"cell_type": "code", "source": "import pyspark\nimport dxdata\nimport dxpy\nimport pandas as pd\nfrom datetime import date, datetime\nimport os \nimport numpy as np\nimport random\nimport shutil\nimport glob\nimport requests\nfrom functools import reduce\n\nsc = pyspark.SparkContext()\nspark = pyspark.sql.SparkSession(sc)\n", "metadata": {"trusted": true, "tags": []}, "execution_count": 1, "outputs": [], "id": "cd14b439-3a34-4910-96d6-fc7afaede06f"}, {"cell_type": "markdown", "source": "## Initialize variables", "metadata": {"tags": []}, "id": "a579174a-8b5a-488d-af45-aa9ed13205d2"}, {"cell_type": "code", "source": "gene_names = [\"CRIM1\"]\n", "metadata": {"trusted": true, "tags": []}, "execution_count": 2, "outputs": [], "id": "adff6de3-eef6-4029-ae05-b3905bf83d43"}, {"cell_type": "markdown", "source": "## Initialize helper functions", "metadata": {"tags": []}, "id": "2efc43ef-80dd-4450-9c64-98735c8a2216"}, {"cell_type": "code", "source": "def fetch_gene_info_ensembl(gene_names, species=\"human\", genome_version=\"GRCh38\"):\n    gene_info_dict = {}\n    server = \"https://rest.ensembl.org\"\n    \n    for gene_name in gene_names:\n        endpoint = f\"/lookup/symbol/{species}/{gene_name}\"\n        headers = {\"Content-Type\": \"application/json\"}\n\n        response = requests.get(server + endpoint, headers=headers, params={\"expand\": \"1\"})\n        if not response.ok:\n            print(f\"Fetching failed for {gene_name}\")\n            continue\n\n        data = response.json()\n        gene_info = {\n            \"gene_name\": data.get(\"display_name\", gene_name),\n            \"chromosome\": f\"chr{data['seq_region_name']}\",\n            \"start\": int(data[\"start\"]),\n            \"end\": int(data[\"end\"]),\n            \"genome_version\": genome_version\n        }\n\n        gene_info_dict[gene_name] = gene_info\n\n    return gene_info_dict\n", "metadata": {"trusted": true, "tags": []}, "execution_count": 3, "outputs": [], "id": "3c7822be-2a15-455a-ab3f-28a8211ab69b"}, {"cell_type": "markdown", "source": "# Fetch participants from each phenotype", "metadata": {"tags": []}, "id": "6db4eb5c-7b62-4cca-9413-3f60389debfa"}, {"cell_type": "markdown", "source": "## Grab the dataset containing participant information", "metadata": {"tags": []}, "id": "4759399f-16f7-4135-b76d-309603968270"}, {"cell_type": "code", "source": "dispensed_dataset_id = dxpy.find_one_data_object(typename=\"Dataset\", name=\"app*.dataset\", folder=\"/\", name_mode=\"glob\")[\"id\"]\ndataset = dxdata.load_dataset(id=dispensed_dataset_id)\nparticipant = dataset[\"participant\"]\n", "metadata": {"trusted": true, "tags": []}, "execution_count": 4, "outputs": [], "id": "cbfbf832-1d99-434a-9390-ede2b7245c42"}, {"cell_type": "markdown", "source": "## Retrieve Cases", "metadata": {"tags": []}, "id": "9ea3457f-503d-4b56-9aed-43ca4255495d"}, {"cell_type": "markdown", "source": "### Pull down the fields we need \nhttps://docs.google.com/document/d/1AebkQ-Nxrk63jhsDzZpn5QD-7EK4unsykHVj-saEm3U/edit?usp=sharing", "metadata": {"tags": []}, "id": "b85518c4-29ea-41cf-88d5-aee787f51bda"}, {"cell_type": "code", "source": "field_names = [\n    \"eid\", \n    \"p31\", \n    \"p34\", \n    \"p21022\", \n    \"p42018\", \n    \"p42020\", \n    \"p42032\", \n    \"p22009_a1\", \n    \"p22009_a2\", \n    \"p22009_a3\", \n    \"p22009_a4\", \n    \"p22009_a5\", \n    \"p40000_i0\",\n]\ndf_cases = participant.retrieve_fields(names=field_names, coding_values=\"replace\", engine=dxdata.connect())\ndf_cases = df_cases.toPandas()\n", "metadata": {"trusted": true, "tags": []}, "execution_count": 5, "outputs": [{"name": "stderr", "text": "/opt/conda/lib/python3.11/site-packages/thrift/transport/TSSLSocket.py:53: DeprecationWarning: ssl.PROTOCOL_TLS is deprecated\n  self._context = ssl.SSLContext(ssl_version)\n", "output_type": "stream"}], "id": "f917dc60-cc2c-4991-9732-5b632c39258f"}, {"cell_type": "markdown", "source": "### Rename columns to be human-readable", "metadata": {"tags": []}, "id": "04b0abf1-b83d-49ae-8513-d95afee0e91a"}, {"cell_type": "code", "source": "df_cases = df_cases.rename(columns={\n    \"eid\":\"ID\",\n    \"p31\":\"GENETIC_SEX\", \n    \"p34\":\"BIRTH_YEAR\", \n    \"p21022\":\"AGE_OF_RECRUIT\",\n    \"p42032\":\"PD_DATE\",\n    \"p22009_a1\":\"PC1\",\n    \"p22009_a2\":\"PC2\",\n    \"p22009_a3\":\"PC3\",\n    \"p22009_a4\":\"PC4\",\n    \"p22009_a5\":\"PC5\",\n    \"p40000_i0\":\"DATE_OF_DEATH\",\n})\n", "metadata": {"trusted": true, "tags": []}, "execution_count": 6, "outputs": [], "id": "f6709363-3dd9-4265-a3c0-324eab83c72f"}, {"cell_type": "markdown", "source": "### Find participants with PD", "metadata": {"tags": []}, "id": "70b05066-05f2-460d-9566-22f8c2d06f01"}, {"cell_type": "code", "source": "df_cases = df_cases[~df_cases[f\"PD_DATE\"].isna()]\ndf_cases = df_cases[[\n    \"ID\", \n    \"GENETIC_SEX\", \n    \"BIRTH_YEAR\", \n    \"AGE_OF_RECRUIT\", \n    \"PD_DATE\", \n    \"PC1\", \n    \"PC2\", \n    \"PC3\", \n    \"PC4\", \n    \"PC5\", \n    \"DATE_OF_DEATH\",\n]]\ndf_cases[\"ID\"] = pd.to_numeric(df_cases[\"ID\"])\n", "metadata": {"trusted": true, "tags": []}, "execution_count": 7, "outputs": [], "id": "4fe6d7fc-01b8-401f-a7f4-584ca57cdeef"}, {"cell_type": "markdown", "source": "## Retrieve Controls", "metadata": {"tags": []}, "id": "eba6b23a-ece6-40c2-9ec1-71b88e992e48"}, {"cell_type": "markdown", "source": "### Retrieve field names of interest for each participant", "metadata": {"tags": []}, "id": "8c8baa9a-cdf5-4aae-bfa3-c2123f2d9f26"}, {"cell_type": "code", "source": "# Date G10 first reported (huntington's disease),\n# Date D11 first reported (hereditary ataxia), \n# Date G12 first reported (spinal muscular atrophy and related syndromes), \n# Date G13 first reported (systemic atrophies primarily affecting central nervous system in diseases classified elswhere), \n# Date G14 first reported (postpolio syndrome), \n# Date G20 first reported (parkinson's disease), \n# Date G21 first reported (secondary parkinsonism), \n# Date G22 first reported (parkinsonism in diseases classified elsewhere), \n# Date G23 first reported (other degenerative diseases of basal ganglia), \n# Date G24 first reported (dystonia), \n# Date G25 first reported (other extrapyramidal and movement disorders), \n# Date G30 first reported (alzheimer's disease), \n# Date G31 first reported (other degenerative diseases of nervous system, not elsewhere classified), \n# Date G32 first reported (other degenerative disorders of nervous system in diseases classified elsewhere), \n# Date G35 first reported (multiple sclerosis), \n# Date G36 first reported (other acute disseminated demyelination), \n# Date G37 first reported (other demyelinating diseases of central nervous system), \n# Date G45 first reported (transient cerebral ischaemic attacks and related syndromes), \n# Date G46 first reported (vascular syndromes of brain in cerebrovascular diseases), \n# Date G50 first reported (disorders of trigeminal nerve), \n# Date G52 first reported (disorders of other cranial nerves), \n# Date G53 first reported (cranial nerve disorders in diseases classified elsewhere), \n# Date G54 first reported (nerve root and plexus disorders), \n# Date G55 first reported (nerve root and plexus compressions in diseases classified elsewhere), \n# Date G56 first reported (mononeuropathies of upper limb), \n# Date G57 first reported (mononeuropathies of lower limb), \n# Date G58 first reported (other mononeuropathies), \n# Date G59 first reported (mononeuropathy in diseases classified elsewhere), \n# Date G60 first reported (hereditary and idiopathic neuropathy), \n# Date G61 first reported (inflammatory polyneuropathy), \n# Date G62 first reported (other polyneuropathies), \n# Date G63 first reported (polyneuropathy in diseases classified elsewhere), \n# Date G64 first reported (other disorders of peripheral nervous system), \n# Date G70 first reported (myasthenia gravis and other myoneural disorders), \n# Date G71 first reported (primary disorders of muscles), \n# Date G72 first reported (other myopathies), \n# Date G73 first reported (disorders of myoneural junction and muscle in diseases classified elsewhere), \n# Date G80 first reported (infantile cerebral palsy), \n# Date G81 first reported (hemiplegia), \n# Date G82 first reported (paraplegia and tetraplegia), \n# Date G83 first reported (other paralytic syndromes), \n# Date G90 first reported (disorders of autonomic nervous system),\n# Date G91 first reported (hydrocephalus), \n# Date G92 first reported (toxic encephalopathy), \n# Date G93 first reported (other disorders of brain), \n# Date G94 first reported (other disorders of brain in diseases classified elsewhere), \n# Date G96 first reported (other disorders of central nervous system), \n# Date G97 first reported (postprocedural disorders of nervous system, not elsewhere classified),  \n# Date G98 first reported (other disorders of nervous system, not elsewhere classified), \n# Date G99 first reported (other disorders of nervous system in diseases classified elsewhere), \n# Date of all cause dementia report, \n# Date of alzheimer's disease report, \n# Date of vascular dementia report, \n# Date of frontotemporal dementia report, \n# Date of motor neurone disease report, \n# Date of all cause parkinsonism report, \n# Date of parkinson's disease report, \n# Date of progressive supranuclear palsy report, \n# Date of multiple system atrophy report, \n# Genetic ethnic grouping, \n# Age at recruitment, \n# Townsend deprivation index at recruitment, \n# Sex, \n# Genetic Principal components | Array 1, \n# Genetic Principal components | Array 2, \n# Genetic Principal components | Array 3, \n# Genetic Principal components | Array 4, \n# Genetic Principal components | Array 5\n\nfield_names = [\"eid\", \"p131012\", \"p131016\", \"p131018\", \"p131020\", \"p131022\", \"p131024\", \"p131026\", \"p131028\", \"p131030\", \"p131036\", \"p131038\", \"p131040\", \n               \"p131042\", \"p131046\", \"p131056\", \"p131058\", \"p131062\", \"p131066\", \"p131068\", \"p131070\", \"p131074\", \"p131076\", \"p131078\", \"p131080\", \"p131082\", \n               \"p131084\", \"p131086\", \"p131088\", \"p131090\", \"p131092\", \"p131094\", \"p131096\", \"p131098\", \"p131100\", \"p131102\", \"p131104\", \"p131106\", \"p131108\", \n               \"p131110\", \"p131112\", \"p131114\", \"p131116\", \"p131120\", \"p131122\", \"p131124\", \"p131126\",  \"p42018\", \"p42020\", \"p42022\", \"p42024\", \"p42028\", \n               \"p42030\", \"p42032\", \"p42034\", \"p42036\", \"p22006\", \"p21022\", \"p31\", \"p22009_a1\", \"p22009_a2\", \"p22009_a3\", \"p22009_a4\", \"p22009_a5\", \"p34\", \n               \"p40000_i0\", \"p20110_i0\", \"p20110_i1\", \"p20110_i2\", \"p20110_i3\", \"p20107_i0\", \"p20107_i1\", \"p20107_i2\", \"p20107_i3\"]\ndf_control = participant.retrieve_fields(names=field_names, coding_values=\"replace\", engine=dxdata.connect())\ndf_control = df_control.toPandas()\n", "metadata": {"trusted": true, "tags": []}, "execution_count": 8, "outputs": [{"name": "stderr", "text": "/opt/conda/lib/python3.11/site-packages/thrift/transport/TSSLSocket.py:53: DeprecationWarning: ssl.PROTOCOL_TLS is deprecated\n  self._context = ssl.SSLContext(ssl_version)\n", "output_type": "stream"}], "id": "e8bfdd82-c980-45fb-9051-8fb3106f1f39"}, {"cell_type": "markdown", "source": "### Remove participants with any of the listed conditions", "metadata": {"tags": []}, "id": "236a6214-b21d-4b93-a9b4-a6e09aa60699"}, {"cell_type": "code", "source": "df_control = df_control[df_control[\"p131012\"].isnull() & df_control[\"p131016\"].isnull() & df_control[\"p131018\"].isnull() & df_control[\"p131020\"].isnull() \n                        & df_control[\"p131022\"].isnull() & df_control[\"p131024\"].isnull() & df_control[\"p131026\"].isnull() & df_control[\"p131028\"].isnull() \n                        & df_control[\"p131030\"].isnull() & df_control[\"p131036\"].isnull() & df_control[\"p131038\"].isnull() & df_control[\"p131040\"].isnull() \n                        & df_control[\"p131042\"].isnull() & df_control[\"p131046\"].isnull() & df_control[\"p131056\"].isnull() & df_control[\"p131058\"].isnull() \n                        & df_control[\"p131062\"].isnull() & df_control[\"p131066\"].isnull() & df_control[\"p131068\"].isnull() & df_control[\"p131070\"].isnull() \n                        & df_control[\"p131074\"].isnull() & df_control[\"p131076\"].isnull() & df_control[\"p131078\"].isnull() & df_control[\"p131080\"].isnull() \n                        & df_control[\"p131082\"].isnull() & df_control[\"p131084\"].isnull() & df_control[\"p131086\"].isnull() & df_control[\"p131088\"].isnull() \n                        & df_control[\"p131090\"].isnull() & df_control[\"p131092\"].isnull() & df_control[\"p131094\"].isnull() & df_control[\"p131096\"].isnull() \n                        & df_control[\"p131098\"].isnull() & df_control[\"p131100\"].isnull() & df_control[\"p131102\"].isnull() & df_control[\"p131104\"].isnull() \n                        & df_control[\"p131106\"].isnull() & df_control[\"p131108\"].isnull() & df_control[\"p131110\"].isnull() & df_control[\"p131112\"].isnull() \n                        & df_control[\"p131114\"].isnull() & df_control[\"p131116\"].isnull() & df_control[\"p131120\"].isnull() & df_control[\"p131122\"].isnull() \n                        & df_control[\"p131124\"].isnull() & df_control[\"p131126\"].isnull() & df_control[\"p42018\"].isnull() & df_control[\"p42020\"].isnull() \n                        & df_control[\"p42022\"].isnull() & df_control[\"p42024\"].isnull() & df_control[\"p42028\"].isnull() & df_control[\"p42030\"].isnull() \n                        & df_control[\"p42032\"].isnull() & df_control[\"p42034\"].isnull() & df_control[\"p42036\"].isnull()]\n", "metadata": {"trusted": true, "tags": []}, "execution_count": 9, "outputs": [], "id": "88d075bb-f716-4006-a411-2d8559db092a"}, {"cell_type": "markdown", "source": "### Remove participants whose parents have AD or PD", "metadata": {"tags": []}, "id": "ee348fde-73f0-42cf-b700-a4d21cf96144"}, {"cell_type": "code", "source": "# Columns defining all instances of parent illness\nparent_illness_cols = [\"p20110_i0\", \"p20110_i1\", \"p20110_i2\", \"p20110_i3\", \"p20107_i0\", \"p20107_i1\", \"p20107_i2\", \"p20107_i3\"]\n\n# Convert None values to empty lists\nfor illness_col in parent_illness_cols:\n    df_control[illness_col] = df_control[illness_col].apply(lambda l: l if isinstance(l, list) else [])\n\n# Define a condition as anybody who has never reported a parent as having AD or PD\ncondition = lambda participant: all((\"Alzheimer's disease/dementia\" not in illnesses and \"Parkinson's disease\" not in illnesses) for illnesses in participant[parent_illness_cols])\n\n# Apply the condition to give all participants who have a parent who has/had AD or PD\ndf_control = df_control[df_control.apply(condition, axis=1)]\n", "metadata": {"trusted": true, "tags": []}, "execution_count": 10, "outputs": [], "id": "940fc55e-e654-48f2-8b05-5e51c9fc64f2"}, {"cell_type": "markdown", "source": "### Remove participants below the defined age threshold", "metadata": {"tags": []}, "id": "5568a2f0-906d-458a-bdc1-1f2b2912ee11"}, {"cell_type": "code", "source": "df_control = df_control[df_control[\"p21022\"] >= 65]\n", "metadata": {"trusted": true, "tags": []}, "execution_count": 11, "outputs": [], "id": "22cc264e-fb68-4847-ac74-6bc16445d431"}, {"cell_type": "markdown", "source": "### Rename columns", "metadata": {"tags": []}, "id": "01cc3aa6-86af-46d2-b865-0631e97dfe1f"}, {"cell_type": "code", "source": "df_control = df_control[[\"eid\", \"p21022\", \"p31\", \"p22009_a1\", \"p22009_a2\", \"p22009_a3\", \"p22009_a4\", \"p22009_a5\", \"p34\", \"p22006\", \"p40000_i0\"]]\ndf_control.rename(columns={\n    \"eid\":\"ID\",\n    \"p21022\":\"AGE_OF_RECRUIT\", \n    \"p31\":\"GENETIC_SEX\", \n    \"p22009_a1\":\"PC1\", \n    \"p22009_a2\":\"PC2\", \n    \"p22009_a3\":\"PC3\", \n    \"p22009_a4\":\"PC4\", \n    \"p22009_a5\":\"PC5\", \n    \"p34\":\"BIRTH_YEAR\", \n    \"p22006\":\"ETHNICITY\", \n    \"p40000_i0\":\"DATE_OF_DEATH\",\n}, inplace=True)\ndf_control[\"ID\"] = pd.to_numeric(df_control[\"ID\"])\ndf_control.info()\n", "metadata": {"trusted": true, "tags": []}, "execution_count": 12, "outputs": [{"name": "stdout", "text": "<class 'pandas.core.frame.DataFrame'>\nIndex: 59611 entries, 38 to 502155\nData columns (total 11 columns):\n #   Column          Non-Null Count  Dtype  \n---  ------          --------------  -----  \n 0   ID              59611 non-null  int64  \n 1   AGE_OF_RECRUIT  59611 non-null  int64  \n 2   GENETIC_SEX     59611 non-null  object \n 3   PC1             57980 non-null  float64\n 4   PC2             57980 non-null  float64\n 5   PC3             57980 non-null  float64\n 6   PC4             57980 non-null  float64\n 7   PC5             57980 non-null  float64\n 8   BIRTH_YEAR      59611 non-null  int64  \n 9   ETHNICITY       50743 non-null  object \n 10  DATE_OF_DEATH   8986 non-null   object \ndtypes: float64(5), int64(3), object(3)\nmemory usage: 5.5+ MB\n", "output_type": "stream"}], "id": "c4fabba8-2442-4fc0-8bb3-ff9acf870d4d"}, {"cell_type": "markdown", "source": "## Get list of IDs for each phenotype", "metadata": {"tags": []}, "id": "23511f34-8311-40ab-bf0e-6f3c84c2900e"}, {"cell_type": "code", "source": "ids_control = df_control[\"ID\"].tolist()\nids_cases = df_cases[\"ID\"].tolist()\n", "metadata": {"trusted": true, "tags": [], "collapsed": true, "jupyter": {"outputs_hidden": true}}, "execution_count": 13, "outputs": [], "id": "ef16630b-f5b9-457d-9dcd-1f6be68f8673"}, {"cell_type": "markdown", "source": "# Remove related individuals", "metadata": {"tags": [], "jp-MarkdownHeadingCollapsed": true}, "id": "e703dfb9-c45f-4f4e-b7c1-5371fcd1f38e"}, {"cell_type": "markdown", "source": "### Fetch relatedness data", "metadata": {"tags": []}, "id": "e75c2575-7728-4f0c-a8a2-c16e4909ed29"}, {"cell_type": "code", "source": "! dx download \"Bulk/Genotype Results/Genotype calls/ukb_rel.dat\"\ndf_full_related = pd.read_csv(\"ukb_rel.dat\", sep = \" \")\ndf_full_related = df_full_related[df_full_related[\"Kinship\"] > 0.0884]\n", "metadata": {"trusted": true, "tags": [], "collapsed": true, "jupyter": {"outputs_hidden": true}}, "execution_count": 16, "outputs": [{"name": "stdout", "text": "\u001b[2K[===========================================================>] Completed 3,806,281 of 3,806,281 bytes (100%) /opt/notebooks/ukb_rel.datt\n", "output_type": "stream"}], "id": "3d9514f1-6ea1-498a-9442-75b43e5b302c"}, {"cell_type": "markdown", "source": "### Keep only rows with both participants in cohorts of interest", "metadata": {"tags": []}, "id": "75f9c846-5b7d-4c2e-aa22-cf56fd9f106b"}, {"cell_type": "code", "source": "df_related_cohort = df_full_related.loc[df_full_related[\"ID1\"].isin(ids_cases + ids_control) & df_full_related[\"ID2\"].isin(ids_cases + ids_control)]\ndf_related_cohort = df_related_cohort.reset_index(drop=True)\n", "metadata": {"trusted": true, "tags": [], "collapsed": true, "jupyter": {"outputs_hidden": true}}, "execution_count": 17, "outputs": [], "id": "858045e4-fe3e-48e0-9181-f8239ae098b6"}, {"cell_type": "markdown", "source": "### Maximize the number of cases included", "metadata": {"tags": []}, "id": "3ac59ac1-0f14-4fef-8b81-407a598870c7"}, {"cell_type": "code", "source": "df_flipped = df_related_cohort[df_related_cohort[\"ID1\"].isin(ids_control) & df_related_cohort[\"ID2\"].isin(ids_cases)].copy()\ndf_related_cohort = df_related_cohort[~(df_related_cohort[\"ID1\"].isin(ids_control) & df_related_cohort[\"ID2\"].isin(ids_cases))]\ndf_flipped.rename(columns={\"ID1\":\"ID2\", \"ID2\":\"ID1\"}, inplace=True)\ndf_related_cohort = pd.concat([df_related_cohort, df_flipped])\n", "metadata": {"trusted": true, "tags": [], "collapsed": true, "jupyter": {"outputs_hidden": true}}, "execution_count": 18, "outputs": [], "id": "d728ac8d-d930-421c-a8e5-f15a91bb2a08"}, {"cell_type": "markdown", "source": "### Get set of participants to remove", "metadata": {"tags": []}, "id": "758e07fe-ec1c-457d-b8ab-d0b678f41017"}, {"cell_type": "code", "source": "ids_to_remove = set(df_related_cohort[\"ID2\"])\nprint(f\"Removing {len(ids_to_remove)} participants\")\n", "metadata": {"trusted": true, "tags": [], "collapsed": true, "jupyter": {"outputs_hidden": true}}, "execution_count": 19, "outputs": [{"name": "stdout", "text": "Removing 737 participants\n", "output_type": "stream"}], "id": "b5dbe38d-6111-47a7-b8cd-04fcd5261643"}, {"cell_type": "markdown", "source": "### Filter ID lists accordingly", "metadata": {"tags": []}, "id": "98554e3c-a3b2-42ca-9a3e-07a3ebbfce22"}, {"cell_type": "code", "source": "ids_cases = [iid for iid in ids_cases if iid not in ids_to_remove]\nids_control = [iid for iid in ids_control if iid not in ids_to_remove]\n", "metadata": {"trusted": true, "tags": [], "collapsed": true, "jupyter": {"outputs_hidden": true}}, "execution_count": 20, "outputs": [], "id": "0963afa9-d662-446d-98ca-d4b62d258a9f"}, {"cell_type": "code", "source": "print(len(ids_cases))\nprint(len(ids_control))\nprint(len([iid for iid in ids_control if iid in ids_cases]))\n", "metadata": {"trusted": true, "tags": [], "collapsed": true, "jupyter": {"outputs_hidden": true}}, "execution_count": 21, "outputs": [{"name": "stdout", "text": "4252\n57135\n0\n", "output_type": "stream"}], "id": "07a2f004-ada2-4256-9b16-55589c87879b"}, {"cell_type": "markdown", "source": "### Save the IDs of each participant to a txt file", "metadata": {"tags": []}, "id": "4384b677-27ec-47f6-813b-2280a0b7205e"}, {"cell_type": "code", "source": "with open(\"cases_ids_pre_VCF.txt\", \"w\") as file:\n    for iid in ids_cases:\n        file.write(f\"{iid}\\n\")\n", "metadata": {"trusted": true, "tags": [], "collapsed": true, "jupyter": {"outputs_hidden": true}}, "execution_count": 22, "outputs": [], "id": "c00fb845-0668-4961-ba0a-623eac074715"}, {"cell_type": "code", "source": "with open(\"control_ids_pre_VCF.txt\", \"w\") as file:\n    for iid in ids_control:\n        file.write(f\"{iid}\\n\")\n", "metadata": {"trusted": true, "tags": [], "collapsed": true, "jupyter": {"outputs_hidden": true}}, "execution_count": 23, "outputs": [], "id": "6a0997d6-3bb3-4106-9e76-49bb4da0fae0"}, {"cell_type": "code", "source": "with open(\"ids_pre_VCF.txt\", \"w\") as file:\n    for iid in ids_cases + ids_control:\n        file.write(f\"{iid}\\n\")\n", "metadata": {"trusted": true, "tags": [], "collapsed": true, "jupyter": {"outputs_hidden": true}}, "execution_count": 24, "outputs": [], "id": "f581ecc9-00d1-4ee5-ad23-ea0c8eb2105e"}, {"cell_type": "markdown", "source": "# Filter out participants without WGS data", "metadata": {"tags": [], "jp-MarkdownHeadingCollapsed": true}, "id": "0873ee2d-bfa5-4cc1-9deb-9eaafde85999"}, {"cell_type": "markdown", "source": "## Only include participants with WGS data", "metadata": {"tags": []}, "id": "7d1a6a8a-8731-468c-b841-db8f2f72e368"}, {"cell_type": "code", "source": "! dx download /data/pvcf_full_ids.txt --overwrite\n! grep -Fwf pvcf_full_ids.txt ids_pre_VCF.txt > filtered_sample_ids.txt\n! grep -Fwf pvcf_full_ids.txt cases_ids_pre_VCF.txt > filtered_cases_ids.txt\n! grep -Fwf pvcf_full_ids.txt control_ids_pre_VCF.txt > filtered_control_ids.txt\n", "metadata": {"trusted": true, "tags": [], "collapsed": true, "jupyter": {"outputs_hidden": true}}, "execution_count": 25, "outputs": [{"name": "stdout", "text": "\u001b[2K[===========================================================>] Completed 3,924,328 of 3,924,328 bytes (100%) /opt/notebooks/pvcf_full_ids.txtt\n", "output_type": "stream"}], "id": "0d7f0a68-4066-404c-91b1-648c96b8fbc6"}, {"cell_type": "code", "source": "with open(\"filtered_cases_ids.txt\", \"r\") as file:\n    ids_cases = [int(line.strip()) for line in file]\nwith open(\"filtered_control_ids.txt\", \"r\") as file:\n    ids_control = [int(line.strip()) for line in file]\n", "metadata": {"trusted": true, "tags": [], "collapsed": true, "jupyter": {"outputs_hidden": true}}, "execution_count": 26, "outputs": [], "id": "618075de-394b-4149-80f3-f0834b7d3a3a"}, {"cell_type": "markdown", "source": "## Get participant IDs for each phenotype", "metadata": {"tags": [], "jp-MarkdownHeadingCollapsed": true}, "id": "1f84405d-6953-402f-8e1b-9af96e3fa536"}, {"cell_type": "code", "source": "df_cases = df_cases[df_cases[\"ID\"].isin(ids_cases)]\ndf_control = df_control[df_control[\"ID\"].isin(ids_control)]\n", "metadata": {"trusted": true, "tags": [], "collapsed": true, "jupyter": {"outputs_hidden": true}}, "execution_count": 27, "outputs": [], "id": "71fe61ad-b1dd-4816-b625-310e0382201a"}, {"cell_type": "code", "source": "print(f\"Number of Cases:                 {len(ids_cases)}\")\nprint(f\"Number of Control participants:  {len(ids_control)}\")\n", "metadata": {"trusted": true, "tags": [], "collapsed": true, "jupyter": {"outputs_hidden": true}}, "execution_count": 28, "outputs": [{"name": "stdout", "text": "Number of Cases:                 4225\nNumber of Control participants:  56819\n", "output_type": "stream"}], "id": "9626b5ed-6274-439d-8a93-33ad761fa874"}, {"cell_type": "code", "source": "! dx upload filtered_sample_ids.txt --path /results/alox15/sample_ids.txt\n! dx upload filtered_cases_ids.txt --path /results/alox15/cases_ids.txt\n! dx upload filtered_control_ids.txt --path /results/alox15/control_ids.txt\n", "metadata": {"trusted": true, "tags": [], "collapsed": true, "jupyter": {"outputs_hidden": true}}, "execution_count": null, "outputs": [], "id": "f4fa38b3-d52e-400b-b7a1-e6d9acdd3213"}, {"cell_type": "markdown", "source": "## Generate pheno and sex files", "metadata": {"tags": [], "jp-MarkdownHeadingCollapsed": true}, "id": "c0364eca-e5ea-47c1-9f90-83304ed22569"}, {"cell_type": "code", "source": "! dx download /results/alox15/sample_ids.txt\n! dx download /results/alox15/cases_ids.txt\n! dx download /results/alox15/control_ids.txt\n", "metadata": {"trusted": true, "tags": [], "collapsed": true, "jupyter": {"outputs_hidden": true}}, "execution_count": 29, "outputs": [{"name": "stdout", "text": "Error: path \"/opt/notebooks/sample_ids.txt\" already exists but -f/--overwrite\nwas not set\nError: path \"/opt/notebooks/cases_ids.txt\" already exists but -f/--overwrite\nwas not set\nError: path \"/opt/notebooks/control_ids.txt\" already exists but -f/--overwrite\nwas not set\n", "output_type": "stream"}], "id": "089ff993-8aa5-4a7a-ad7b-6f489b6e4910"}, {"cell_type": "code", "source": "with open('cases_ids.txt', 'r') as file:\n    ids_cases = file.read().splitlines()\nwith open('control_ids.txt', 'r') as file:\n    ids_control = file.read().splitlines()\nwith open('sample_ids.txt', 'r') as file:\n    ids_combined = file.read().splitlines()\n    ", "metadata": {"trusted": true, "tags": [], "collapsed": true, "jupyter": {"outputs_hidden": true}}, "execution_count": 30, "outputs": [], "id": "958929bf-ea63-425b-97f9-696b2d41a751"}, {"cell_type": "code", "source": "cases_genetic_sex = [1 if df_cases[\"GENETIC_SEX\"][df_cases[\"ID\"] == int(iid)].values[0] == \"Male\" else 2 for iid in ids_cases]\ncontrol_genetic_sex = [1 if df_control[\"GENETIC_SEX\"][df_control[\"ID\"] == int(iid)].values[0] == \"Male\" else 2 for iid in ids_control]\ngenetic_sex = cases_genetic_sex + control_genetic_sex\n\npheno = [2] * len(ids_cases) + [1] * len(ids_control)\n", "metadata": {"trusted": true, "tags": [], "collapsed": true, "jupyter": {"outputs_hidden": true}}, "execution_count": 44, "outputs": [], "id": "bbc4b659-547a-4964-9bea-f6b4fe33d09e"}, {"cell_type": "code", "source": "with open('sex.txt', 'w') as f:\n    for line in zip(ids_combined, genetic_sex):\n        f.write(f\"{line[0]}\\t{line[0]}\\t{line[1]}\\n\")\n        \nwith open('pheno.txt', 'w') as f:\n    for line in zip(ids_combined, pheno):\n        f.write(f\"{line[0]}\\t{line[0]}\\t{line[1]}\\n\")\n", "metadata": {"trusted": true, "tags": [], "collapsed": true, "jupyter": {"outputs_hidden": true}}, "execution_count": 46, "outputs": [], "id": "330e2bcf-822d-4e5a-bf21-6a57d25209a2"}, {"cell_type": "code", "source": "! dx upload sex.txt --path /results/alox15/sex.txt\n! dx upload pheno.txt --path /results/alox15/pheno.txt\n", "metadata": {"trusted": true, "tags": [], "collapsed": true, "jupyter": {"outputs_hidden": true}}, "execution_count": 47, "outputs": [{"name": "stdout", "text": "[===========================================================>] Uploaded 1,098,792 of 1,098,792 bytes (100%) sex.txt\nID                                file-Gv22ZFQJf68V6Bgq737qfJ2Y\nClass                             file\nProject                           project-Gpz01Y8Jf68b52JGvVXjbQp1\nFolder                            /results/alox15\nName                              sex.txt\nState                             \u001b[33mclosing\u001b[0m\nVisibility                        visible\nTypes                             -\nProperties                        -\nTags                              -\nOutgoing links                    -\nCreated                           Mon Oct  7 18:36:30 2024\nCreated by                        spencermg3\n via the job                      job-Gv1y880Jf68z0vG05gyvY068\nLast modified                     Mon Oct  7 18:36:30 2024\nMedia type                        \narchivalState                     \"live\"\ncloudAccount                      \"cloudaccount-dnanexus\"\n[===========================================================>] Uploaded 1,098,792 of 1,098,792 bytes (100%) pheno.txt\nID                                file-Gv22ZFjJf68XqJqQxkkzP6Y7\nClass                             file\nProject                           project-Gpz01Y8Jf68b52JGvVXjbQp1\nFolder                            /results/alox15\nName                              pheno.txt\nState                             \u001b[33mclosing\u001b[0m\nVisibility                        visible\nTypes                             -\nProperties                        -\nTags                              -\nOutgoing links                    -\nCreated                           Mon Oct  7 18:36:31 2024\nCreated by                        spencermg3\n via the job                      job-Gv1y880Jf68z0vG05gyvY068\nLast modified                     Mon Oct  7 18:36:32 2024\nMedia type                        \narchivalState                     \"live\"\ncloudAccount                      \"cloudaccount-dnanexus\"\n", "output_type": "stream"}], "id": "a1cb13ec-98fc-4e8d-a1ff-212896cb25b9"}, {"cell_type": "markdown", "source": "# Save and print cohort statistics", "metadata": {"tags": [], "jp-MarkdownHeadingCollapsed": true}, "id": "c96a78b8-db32-4e8f-937f-407092475a68"}, {"cell_type": "code", "source": "df_control.to_csv(\"control.csv\", header=True, index=False)\ndf_cases.to_csv(\"cases.csv\", header=True, index=False)\n", "metadata": {"trusted": true, "tags": [], "collapsed": true, "jupyter": {"outputs_hidden": true}}, "execution_count": null, "outputs": [], "id": "13af93db-5aa4-4da2-8255-fa07c5e48b75"}, {"cell_type": "code", "source": "! dx upload control.csv --path /results/alox15/control.csv\n! dx upload cases.csv --path /results/alox15/cases.csv\n", "metadata": {"trusted": true, "tags": [], "collapsed": true, "jupyter": {"outputs_hidden": true}}, "execution_count": null, "outputs": [], "id": "2eaf30ad-fb5b-4f58-8463-53da85f58a4c"}, {"cell_type": "code", "source": "print(df_control[\"GENETIC_SEX\"].value_counts())\nprint(df_cases[\"GENETIC_SEX\"].value_counts())\nprint(\"\\n\")\n\nprint(f'{df_control[df_control[\"GENETIC_SEX\"] == \"Male\"][\"AGE_OF_RECRUIT\"].mean()} +/- {df_control[df_control[\"GENETIC_SEX\"] == \"Male\"][\"AGE_OF_RECRUIT\"].std()}')\nprint(f'{df_cases[df_cases[\"GENETIC_SEX\"] == \"Male\"][\"AGE_OF_RECRUIT\"].mean()} +/- {df_cases[df_cases[\"GENETIC_SEX\"] == \"Male\"][\"AGE_OF_RECRUIT\"].std()}')\nprint(f'{df_control[df_control[\"GENETIC_SEX\"] == \"Female\"][\"AGE_OF_RECRUIT\"].mean()} +/- {df_control[df_control[\"GENETIC_SEX\"] == \"Female\"][\"AGE_OF_RECRUIT\"].std()}')\nprint(f'{df_cases[df_cases[\"GENETIC_SEX\"] == \"Female\"][\"AGE_OF_RECRUIT\"].mean()} +/- {df_cases[df_cases[\"GENETIC_SEX\"] == \"Female\"][\"AGE_OF_RECRUIT\"].std()}')\nprint(\"\\n\")\n", "metadata": {"trusted": true, "tags": [], "collapsed": true, "jupyter": {"outputs_hidden": true}}, "execution_count": null, "outputs": [], "id": "0ae940a7-abb4-43b2-93b4-b6d07760838a"}, {"cell_type": "markdown", "source": "# Fetch pVCF chunks for ALOX15", "metadata": {"tags": [], "jp-MarkdownHeadingCollapsed": true}, "id": "cfc4bc7c-09fd-412d-88b2-9435b141e066"}, {"cell_type": "code", "source": "gene_info = fetch_gene_info_ensembl(gene_names)\nprint(gene_info)\nfor gene_name in gene_names:\n    start = gene_info[gene_name][\"start\"] // 20000\n    end = fetch_gene_info_ensembl(gene_names)[gene_name][\"end\"] // 20000 + 1\n    print(f\"Chromosome:    {fetch_gene_info_ensembl(gene_names)[gene_name]['chromosome']}\")\n    print(f\"Start b-val:   {start}\")\n    print(f\"End b-val:     {end}\")\n", "metadata": {"trusted": true, "tags": []}, "execution_count": 18, "outputs": [{"traceback": ["\u001b[0;31m---------------------------------------------------------------------------\u001b[0m", "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)", "Cell \u001b[0;32mIn[18], line 1\u001b[0m\n\u001b[0;32m----> 1\u001b[0m gene_info \u001b[38;5;241m=\u001b[39m \u001b[43mfetch_gene_info_ensembl\u001b[49m(gene_names)\n\u001b[1;32m      2\u001b[0m \u001b[38;5;28mprint\u001b[39m(gene_info)\n\u001b[1;32m      3\u001b[0m \u001b[38;5;28;01mfor\u001b[39;00m gene_name \u001b[38;5;129;01min\u001b[39;00m gene_names:\n", "\u001b[0;31mNameError\u001b[0m: name 'fetch_gene_info_ensembl' is not defined"], "ename": "NameError", "evalue": "name 'fetch_gene_info_ensembl' is not defined", "output_type": "error"}], "id": "889aaefc-0f7d-4709-b440-5e0a642a5560"}, {"cell_type": "code", "source": "!dx download ADRD_noFTD_noPARKINSONISM_noALS_noVD_noHT_noCJD_noPDD.CTRL_inclProxy_60older.plink_pheno.txt", "metadata": {"trusted": true, "tags": []}, "execution_count": 15, "outputs": [{"name": "stdout", "text": "\u001b[2K[===========================================================>] Completed 1,634,665 of 1,634,665 bytes (100%) /opt/notebooks/ADRD_noFTD_noPARKINSONISM_noALS_noVD_noHT_noCJD_noPDD.CTRL_inclProxy_60older.plink_pheno.txtt\n", "output_type": "stream"}], "id": "e96270a8-9213-46ca-8572-1e30fec31772"}, {"cell_type": "code", "source": "!head Pheno_ADRD.txt", "metadata": {"trusted": true, "tags": []}, "execution_count": 26, "outputs": [{"name": "stdout", "text": "\"\"\n1010117\n1012224\n1013262\n1015072\n1016036\n1028356\n1029089\n1032991\n1043268\n", "output_type": "stream"}], "id": "1a75e911-02d1-473c-97b0-892df3c42ccc"}, {"cell_type": "code", "source": "input_file = \"Pheno_ADRD.txt\"  # Replace with your file name\noutput_file = \"Pheno_ADRD1.txt\"  # Name of the modified file\n\n# Open the input file and process the header\nwith open(input_file, \"r\") as infile, open(output_file, \"w\") as outfile:\n    # Read the first line (header)\n    header = infile.readline()\n    # Remove double quotes from the header\n    header = header.replace('\"', '')\n    # Write the modified header to the output file\n    outfile.write(header)\n    \n    # Write the remaining lines as they are\n    for line in infile:\n        outfile.write(line)\n\nprint(f\"Modified file saved to: {output_file}\")\n", "metadata": {"trusted": true, "tags": []}, "execution_count": 27, "outputs": [{"name": "stdout", "text": "Modified file saved to: Pheno_ADRD1.txt\n", "output_type": "stream"}], "id": "10b563fe-cec9-48cc-bc24-d6320d46ae86"}, {"cell_type": "code", "source": "!dx download Pheno_ADRD1.txt", "metadata": {"trusted": true, "tags": []}, "execution_count": 3, "outputs": [{"name": "stdout", "text": "Error: path \"/opt/notebooks/Pheno_ADRD1.txt\" already exists but -f/--overwrite\nwas not set\n", "output_type": "stream"}], "id": "09423b75-90b1-4e71-b468-7a100efbe5f9"}, {"cell_type": "code", "source": "%%bash\nfor b_val in {1817..1828};\ndo\n    dx run swiss-army-knife \\\n    -iin=\"/Bulk/DRAGEN\\ WGS/DRAGEN\\ population\\ level\\ WGS\\ variants,\\ pVCF\\ format\\ [500k\\ release]/chr2/ukb24310_c2_b${b_val}_v1.vcf.gz\" \\\n    -iin=\"/Bulk/DRAGEN\\ WGS/DRAGEN\\ population\\ level\\ WGS\\ variants,\\ pVCF\\ format\\ [500k\\ release]/chr2/ukb24310_c2_b${b_val}_v1.vcf.gz.tbi\" \\\n    -iin=\"Pheno_ADRD1.txt\" \\\n    -icmd=\"bcftools view -O z -S Pheno_ADRD1.txt ukb24310_c2_b${b_val}_v1.vcf.gz --force-samples -o CRIM1_b${b_val}.vcf.gz\" \\\n    --instance-type mem2_ssd1_v2_x32 \\\n    --destination \"${projectid}:/01_pvcf_chunks\"\ndone", "metadata": {"trusted": true, "tags": []}, "execution_count": null, "outputs": [{"name": "stdout", "text": "\nUsing input JSON:\n{\n    \"cmd\": \"bcftools view -O z -S Pheno_ADRD1.txt ukb24310_c2_b1817_v1.vcf.gz --force-samples -o CRIM1_b1817.vcf.gz\",\n    \"in\": [\n        {\n            \"$dnanexus_link\": {\n                \"project\": \"project-GkYf2zQJbP2Q3vFgf14863Gf\",\n                \"id\": \"file-GZZbFV0J66GBY1ygJ2Gx7GBx\"\n            }\n        },\n        {\n            \"$dnanexus_link\": {\n                \"project\": \"project-GkYf2zQJbP2Q3vFgf14863Gf\",\n                \"id\": \"file-GZZbGFQJ66GPFpqz2Y39yPPY\"\n            }\n        },\n        {\n            \"$dnanexus_link\": {\n                \"project\": \"project-GkYf2zQJbP2Q3vFgf14863Gf\",\n                \"id\": \"file-GyBYp40JbP2vK42XKPfP37y3\"\n            }\n        }\n    ]\n}\n\nCalling app-Gx52kJQ9GVfXJK4Zpk9x4ZJf with output destination\n  project-GkYf2zQJbP2Q3vFgf14863Gf:/01_pvcf_chunks\n\nJob ID: job-GyBYz2jJbP2bp56XgfGPF1vZ\n\nUsing input JSON:\n{\n    \"cmd\": \"bcftools view -O z -S Pheno_ADRD1.txt ukb24310_c2_b1818_v1.vcf.gz --force-samples -o CRIM1_b1818.vcf.gz\",\n    \"in\": [\n        {\n            \"$dnanexus_link\": {\n                \"project\": \"project-GkYf2zQJbP2Q3vFgf14863Gf\",\n                \"id\": \"file-GZZbFbjJV23KG6Jv1PpygBQX\"\n            }\n        },\n        {\n            \"$dnanexus_link\": {\n                \"project\": \"project-GkYf2zQJbP2Q3vFgf14863Gf\",\n                \"id\": \"file-GZZbGQjJV2342G8q2Q6BVqXy\"\n            }\n        },\n        {\n            \"$dnanexus_link\": {\n                \"project\": \"project-GkYf2zQJbP2Q3vFgf14863Gf\",\n                \"id\": \"file-GyBYp40JbP2vK42XKPfP37y3\"\n            }\n        }\n    ]\n}\n\nCalling app-Gx52kJQ9GVfXJK4Zpk9x4ZJf with output destination\n  project-GkYf2zQJbP2Q3vFgf14863Gf:/01_pvcf_chunks\n\nJob ID: job-GyBYz30JbP2bp56XgfGPF1vf\n\nUsing input JSON:\n{\n    \"cmd\": \"bcftools view -O z -S Pheno_ADRD1.txt ukb24310_c2_b1819_v1.vcf.gz --force-samples -o CRIM1_b1819.vcf.gz\",\n    \"in\": [\n        {\n            \"$dnanexus_link\": {\n                \"project\": \"project-GkYf2zQJbP2Q3vFgf14863Gf\",\n                \"id\": \"file-GZZbFgjJV67QyPv1bB4393vF\"\n            }\n        },\n        {\n            \"$dnanexus_link\": {\n                \"project\": \"project-GkYf2zQJbP2Q3vFgf14863Gf\",\n                \"id\": \"file-GZZbGbQJV67y0vG8p3X0bQ1V\"\n            }\n        },\n        {\n            \"$dnanexus_link\": {\n                \"project\": \"project-GkYf2zQJbP2Q3vFgf14863Gf\",\n                \"id\": \"file-GyBYp40JbP2vK42XKPfP37y3\"\n            }\n        }\n    ]\n}\n\nCalling app-Gx52kJQ9GVfXJK4Zpk9x4ZJf with output destination\n  project-GkYf2zQJbP2Q3vFgf14863Gf:/01_pvcf_chunks\n\nJob ID: job-GyBYz38JbP2pZ9ykVvbQP3gb\n\nUsing input JSON:\n{\n    \"cmd\": \"bcftools view -O z -S Pheno_ADRD1.txt ukb24310_c2_b1820_v1.vcf.gz --force-samples -o CRIM1_b1820.vcf.gz\",\n    \"in\": [\n        {\n            \"$dnanexus_link\": {\n                \"project\": \"project-GkYf2zQJbP2Q3vFgf14863Gf\",\n                \"id\": \"file-GZZbFq0J6JB0YZG4jPPXVxBF\"\n            }\n        },\n        {\n            \"$dnanexus_link\": {\n                \"project\": \"project-GkYf2zQJbP2Q3vFgf14863Gf\",\n                \"id\": \"file-GZZbGb8J6JBF5j1K0Pk7bV5B\"\n            }\n        },\n        {\n            \"$dnanexus_link\": {\n                \"project\": \"project-GkYf2zQJbP2Q3vFgf14863Gf\",\n                \"id\": \"file-GyBYp40JbP2vK42XKPfP37y3\"\n            }\n        }\n    ]\n}\n\nCalling app-Gx52kJQ9GVfXJK4Zpk9x4ZJf with output destination\n  project-GkYf2zQJbP2Q3vFgf14863Gf:/01_pvcf_chunks\n\nJob ID: job-GyBYz3QJbP2pZ9ykVvbQP3gg\n\nUsing input JSON:\n{\n    \"cmd\": \"bcftools view -O z -S Pheno_ADRD1.txt ukb24310_c2_b1821_v1.vcf.gz --force-samples -o CRIM1_b1821.vcf.gz\",\n    \"in\": [\n        {\n            \"$dnanexus_link\": {\n                \"project\": \"project-GkYf2zQJbP2Q3vFgf14863Gf\",\n                \"id\": \"file-GZZbFp8JVXkK9xx0QVJ1zFyp\"\n            }\n        },\n        {\n            \"$dnanexus_link\": {\n                \"project\": \"project-GkYf2zQJbP2Q3vFgf14863Gf\",\n                \"id\": \"file-GZZbGYjJVXk7VYg8zb2vp895\"\n            }\n        },\n        {\n            \"$dnanexus_link\": {\n                \"project\": \"project-GkYf2zQJbP2Q3vFgf14863Gf\",\n                \"id\": \"file-GyBYp40JbP2vK42XKPfP37y3\"\n            }\n        }\n    ]\n}\n\nCalling app-Gx52kJQ9GVfXJK4Zpk9x4ZJf with output destination\n  project-GkYf2zQJbP2Q3vFgf14863Gf:/01_pvcf_chunks\n\nJob ID: job-GyBYz3jJbP2xvq2Z5PJVB6q6\n\nUsing input JSON:\n{\n    \"cmd\": \"bcftools view -O z -S Pheno_ADRD1.txt ukb24310_c2_b1822_v1.vcf.gz --force-samples -o CRIM1_b1822.vcf.gz\",\n    \"in\": [\n        {\n            \"$dnanexus_link\": {\n                \"project\": \"project-GkYf2zQJbP2Q3vFgf14863Gf\",\n                \"id\": \"file-GZZbFxQJz5ZxzzF6kKF1xgzV\"\n            }\n        },\n        {\n            \"$dnanexus_link\": {\n                \"project\": \"project-GkYf2zQJbP2Q3vFgf14863Gf\",\n                \"id\": \"file-GZZbGjQJz5ZzbXXxpk443Z4f\"\n            }\n        },\n        {\n            \"$dnanexus_link\": {\n                \"project\": \"project-GkYf2zQJbP2Q3vFgf14863Gf\",\n                \"id\": \"file-GyBYp40JbP2vK42XKPfP37y3\"\n            }\n        }\n    ]\n}\n\nCalling app-Gx52kJQ9GVfXJK4Zpk9x4ZJf with output destination\n  project-GkYf2zQJbP2Q3vFgf14863Gf:/01_pvcf_chunks\n\nJob ID: job-GyBYz48JbP2g7Q7BKvZXbf80\n\nUsing input JSON:\n{\n    \"cmd\": \"bcftools view -O z -S Pheno_ADRD1.txt ukb24310_c2_b1823_v1.vcf.gz --force-samples -o CRIM1_b1823.vcf.gz\",\n    \"in\": [\n        {\n            \"$dnanexus_link\": {\n                \"project\": \"project-GkYf2zQJbP2Q3vFgf14863Gf\",\n                \"id\": \"file-GZZbFzjJZVpB2BJVb0Px9xzj\"\n            }\n        },\n        {\n            \"$dnanexus_link\": {\n                \"project\": \"project-GkYf2zQJbP2Q3vFgf14863Gf\",\n                \"id\": \"file-GZZbGq8JZVp5Q136p566xvb1\"\n            }\n        },\n        {\n            \"$dnanexus_link\": {\n                \"project\": \"project-GkYf2zQJbP2Q3vFgf14863Gf\",\n                \"id\": \"file-GyBYp40JbP2vK42XKPfP37y3\"\n            }\n        }\n    ]\n}\n\nCalling app-Gx52kJQ9GVfXJK4Zpk9x4ZJf with output destination\n  project-GkYf2zQJbP2Q3vFgf14863Gf:/01_pvcf_chunks\n\nJob ID: job-GyBYz4QJbP2f25JBbF7B6QV2\n\nUsing input JSON:\n{\n    \"cmd\": \"bcftools view -O z -S Pheno_ADRD1.txt ukb24310_c2_b1824_v1.vcf.gz --force-samples -o CRIM1_b1824.vcf.gz\",\n    \"in\": [\n        {\n            \"$dnanexus_link\": {\n                \"project\": \"project-GkYf2zQJbP2Q3vFgf14863Gf\",\n                \"id\": \"file-GZZbG28JZYFgB6bzBGkp7Pzp\"\n            }\n        },\n        {\n            \"$dnanexus_link\": {\n                \"project\": \"project-GkYf2zQJbP2Q3vFgf14863Gf\",\n                \"id\": \"file-GZZbGq8JZYFX2JQBKQ1pVgX9\"\n            }\n        },\n        {\n            \"$dnanexus_link\": {\n                \"project\": \"project-GkYf2zQJbP2Q3vFgf14863Gf\",\n                \"id\": \"file-GyBYp40JbP2vK42XKPfP37y3\"\n            }\n        }\n    ]\n}\n\nCalling app-Gx52kJQ9GVfXJK4Zpk9x4ZJf with output destination\n  project-GkYf2zQJbP2Q3vFgf14863Gf:/01_pvcf_chunks\n\nJob ID: job-GyBYz4jJbP2f25JBbF7B6QV4\n\nUsing input JSON:\n{\n    \"cmd\": \"bcftools view -O z -S Pheno_ADRD1.txt ukb24310_c2_b1825_v1.vcf.gz --force-samples -o CRIM1_b1825.vcf.gz\",\n    \"in\": [\n        {\n            \"$dnanexus_link\": {\n                \"project\": \"project-GkYf2zQJbP2Q3vFgf14863Gf\",\n                \"id\": \"file-GZZbGK8JX4693B2bQzj1k924\"\n            }\n        },\n        {\n            \"$dnanexus_link\": {\n                \"project\": \"project-GkYf2zQJbP2Q3vFgf14863Gf\",\n                \"id\": \"file-GZZbJ70JX465vpyKVfgj018k\"\n            }\n        },\n        {\n            \"$dnanexus_link\": {\n                \"project\": \"project-GkYf2zQJbP2Q3vFgf14863Gf\",\n                \"id\": \"file-GyBYp40JbP2vK42XKPfP37y3\"\n            }\n        }\n    ]\n}\n\nCalling app-Gx52kJQ9GVfXJK4Zpk9x4ZJf with output destination\n  project-GkYf2zQJbP2Q3vFgf14863Gf:/01_pvcf_chunks\n\nJob ID: job-GyBYz50JbP2xvq2Z5PJVB6qB\n\nUsing input JSON:\n{\n    \"cmd\": \"bcftools view -O z -S Pheno_ADRD1.txt ukb24310_c2_b1826_v1.vcf.gz --force-samples -o CRIM1_b1826.vcf.gz\",\n    \"in\": [\n        {\n            \"$dnanexus_link\": {\n                \"project\": \"project-GkYf2zQJbP2Q3vFgf14863Gf\",\n                \"id\": \"file-GZZbGFQJKBJp3z0k4bjPy09J\"\n            }\n        },\n        {\n            \"$dnanexus_link\": {\n                \"project\": \"project-GkYf2zQJbP2Q3vFgf14863Gf\",\n                \"id\": \"file-GZZbJ5jJKBJZ5vQxK7QjBVzG\"\n            }\n        },\n        {\n            \"$dnanexus_link\": {\n                \"project\": \"project-GkYf2zQJbP2Q3vFgf14863Gf\",\n                \"id\": \"file-GyBYp40JbP2vK42XKPfP37y3\"\n            }\n        }\n    ]\n}\n\nCalling app-Gx52kJQ9GVfXJK4Zpk9x4ZJf with output destination\n  project-GkYf2zQJbP2Q3vFgf14863Gf:/01_pvcf_chunks\n\nJob ID: job-GyBYz58JbP2pZ9ykVvbQP3gp\n", "output_type": "stream"}], "id": "5106ceff-1566-4d98-b5ee-302e89494c10"}, {"cell_type": "markdown", "source": "# Combine pVCF chunks into one file", "metadata": {"tags": [], "jp-MarkdownHeadingCollapsed": true}, "id": "5a97d311-e969-43c6-b2c8-eacff5b473d3"}, {"cell_type": "code", "source": "\"\"\"\n------------------------------------------------------------------------------------------------------\n------------------------------------------------------------------------------------------------------\n------------------------------------------------------------------------------------------------------\n------------------------------ PAUSE HERE UNTIL PREVIOUS STEP COMPLETES ------------------------------\n------------------------------------------------------------------------------------------------------\n------------------------------------------------------------------------------------------------------\n------------------------------------------------------------------------------------------------------\n\"\"\"", "metadata": {"collapsed": true, "jupyter": {"outputs_hidden": true}}, "execution_count": null, "outputs": [], "id": "4ed1146d-aaee-4cd5-b100-3b31c40cb52b"}, {"cell_type": "code", "source": "%%bash\ndx run swiss-army-knife \\\n-iin=\"/01_pvcf_chunks/CRIM1_b1817.vcf.gz\" \\\n-iin=\"/01_pvcf_chunks/CRIM1_b1818.vcf.gz\" \\\n-iin=\"/01_pvcf_chunks/CRIM1_b1819.vcf.gz\" \\\n-iin=\"/01_pvcf_chunks/CRIM1_b1820.vcf.gz\" \\\n-iin=\"/01_pvcf_chunks/CRIM1_b1821.vcf.gz\" \\\n-iin=\"/01_pvcf_chunks/CRIM1_b1822.vcf.gz\" \\\n-iin=\"/01_pvcf_chunks/CRIM1_b1823.vcf.gz\" \\\n-iin=\"/01_pvcf_chunks/CRIM1_b1824.vcf.gz\" \\\n-iin=\"/01_pvcf_chunks/CRIM1_b1825.vcf.gz\" \\\n-iin=\"/01_pvcf_chunks/CRIM1_b1826.vcf.gz\" \\\n-iin=\"/01_pvcf_chunks/CRIM1_b1827.vcf.gz\" \\\n-iin=\"/01_pvcf_chunks/CRIM1_b1828.vcf.gz\" \\\n-icmd=\"bcftools concat -O v CRIM1_b1817.vcf.gz CRIM1_b1818.vcf.gz CRIM1_b1819.vcf.gz CRIM1_b1820.vcf.gz CRIM1_b1821.vcf.gz CRIM1_b1822.vcf.gz CRIM1_b1823.vcf.gz CRIM1_b1824.vcf.gz CRIM1_b1825.vcf.gz CRIM1_b1826.vcf.gz CRIM1_b1827.vcf.gz CRIM1_b1828.vcf.gz -o CRIM1.vcf\" \\\n--instance-type mem2_ssd1_v2_x32 \\\n--destination \"${projectid}:/01_pvcf_chunks\"", "metadata": {"trusted": true, "tags": []}, "execution_count": 2, "outputs": [{"name": "stdout", "text": "\nUsing input JSON:\n{\n    \"cmd\": \"bcftools concat -O v CRIM1_b1817.vcf.gz CRIM1_b1818.vcf.gz CRIM1_b1819.vcf.gz CRIM1_b1820.vcf.gz CRIM1_b1821.vcf.gz CRIM1_b1822.vcf.gz CRIM1_b1823.vcf.gz CRIM1_b1824.vcf.gz CRIM1_b1825.vcf.gz CRIM1_b1826.vcf.gz CRIM1_b1827.vcf.gz CRIM1_b1828.vcf.gz -o CRIM1.vcf\",\n    \"in\": [\n        {\n            \"$dnanexus_link\": {\n                \"project\": \"project-GkYf2zQJbP2Q3vFgf14863Gf\",\n                \"id\": \"file-GyBZG6QJ1vKGK42XKPfP39pY\"\n            }\n        },\n        {\n            \"$dnanexus_link\": {\n                \"project\": \"project-GkYf2zQJbP2Q3vFgf14863Gf\",\n                \"id\": \"file-GyBZBZQJfQkGgV6fVq5Z85bJ\"\n            }\n        },\n        {\n            \"$dnanexus_link\": {\n                \"project\": \"project-GkYf2zQJbP2Q3vFgf14863Gf\",\n                \"id\": \"file-GyBZFKjJXxZvf2p6p68XYXJg\"\n            }\n        },\n        {\n            \"$dnanexus_link\": {\n                \"project\": \"project-GkYf2zQJbP2Q3vFgf14863Gf\",\n                \"id\": \"file-GyBZFG0JFJ7vK42XKPfP39K1\"\n            }\n        },\n        {\n            \"$dnanexus_link\": {\n                \"project\": \"project-GkYf2zQJbP2Q3vFgf14863Gf\",\n                \"id\": \"file-GyBZFBQJ2QV4jB6j3vXg4Zzg\"\n            }\n        },\n        {\n            \"$dnanexus_link\": {\n                \"project\": \"project-GkYf2zQJbP2Q3vFgf14863Gf\",\n                \"id\": \"file-GyBZF88Jf1QXx5FBG315Zj3B\"\n            }\n        },\n        {\n            \"$dnanexus_link\": {\n                \"project\": \"project-GkYf2zQJbP2Q3vFgf14863Gf\",\n                \"id\": \"file-GyBZFFjJ8QyXPzPPj5zBJpbP\"\n            }\n        },\n        {\n            \"$dnanexus_link\": {\n                \"project\": \"project-GkYf2zQJbP2Q3vFgf14863Gf\",\n                \"id\": \"file-GyBZBqjJ39Y9fzf531Kffqjj\"\n            }\n        },\n        {\n            \"$dnanexus_link\": {\n                \"project\": \"project-GkYf2zQJbP2Q3vFgf14863Gf\",\n                \"id\": \"file-GyBZBJ8J26vx7x9j0pp7PF6x\"\n            }\n        },\n        {\n            \"$dnanexus_link\": {\n                \"project\": \"project-GkYf2zQJbP2Q3vFgf14863Gf\",\n                \"id\": \"file-GyBZFfQJvG7vy4PjBJBYj9GG\"\n            }\n        },\n        {\n            \"$dnanexus_link\": {\n                \"project\": \"project-GkYf2zQJbP2Q3vFgf14863Gf\",\n                \"id\": \"file-GyBZGGQJx1FQ8Pyxx7bPyB6K\"\n            }\n        },\n        {\n            \"$dnanexus_link\": {\n                \"project\": \"project-GkYf2zQJbP2Q3vFgf14863Gf\",\n                \"id\": \"file-GyBZGF0JV56Xb97486g01Jqy\"\n            }\n        }\n    ]\n}\n\nCalling app-Gx52kJQ9GVfXJK4Zpk9x4ZJf with output destination\n  project-GkYf2zQJbP2Q3vFgf14863Gf:/01_pvcf_chunks\n\nJob ID: job-GyBZj40JbP2xvq2Z5PJVBFXF\n", "output_type": "stream"}], "id": "f99872b4-090e-4c15-8db4-706eb2a7144f"}, {"cell_type": "markdown", "source": "# Normalize VCF before annotation", "metadata": {"tags": [], "jp-MarkdownHeadingCollapsed": true}, "id": "434524be-1789-4c78-8e57-99f2e6de3b0e"}, {"cell_type": "markdown", "source": "## Split multiallelic sites into biallelic records", "metadata": {"tags": []}, "id": "cb462409-8441-4154-b27f-af2f460fa2fc"}, {"cell_type": "code", "source": "\"\"\"\n------------------------------------------------------------------------------------------------------\n------------------------------------------------------------------------------------------------------\n------------------------------------------------------------------------------------------------------\n------------------------------ PAUSE HERE UNTIL PREVIOUS STEP COMPLETES ------------------------------\n------------------------------------------------------------------------------------------------------\n------------------------------------------------------------------------------------------------------\n------------------------------------------------------------------------------------------------------\n\"\"\"", "metadata": {"collapsed": true, "jupyter": {"outputs_hidden": true}}, "execution_count": null, "outputs": [], "id": "c0e635cb-8334-428d-800b-d535ab7c9641"}, {"cell_type": "code", "source": "%%bash\n\ndx run swiss-army-knife \\\n-iin=\"/01_pvcf_chunks/CRIM1.vcf\" \\\n-icmd=\"bcftools norm -m-both -o biallelic.vcf CRIM1.vcf\" \\\n--instance-type mem2_ssd1_v2_x32 \\\n--destination \"${projectid}:/03_normalized\"\n", "metadata": {"trusted": true, "tags": []}, "execution_count": 6, "outputs": [{"name": "stdout", "text": "\nUsing input JSON:\n{\n    \"cmd\": \"bcftools norm -m-both -o biallelic.vcf CRIM1.vcf\",\n    \"in\": [\n        {\n            \"$dnanexus_link\": {\n                \"project\": \"project-GkYf2zQJbP2Q3vFgf14863Gf\",\n                \"id\": \"file-GyBZzXQJx60gbV2ZjBqP22k5\"\n            }\n        }\n    ]\n}\n\nCalling app-Gx52kJQ9GVfXJK4Zpk9x4ZJf with output destination\n  project-GkYf2zQJbP2Q3vFgf14863Gf:/03_normalized\n\nJob ID: job-GyBb5Q8JbP2bx5xbgvKB8GfV\n", "output_type": "stream"}], "id": "56415203-db4b-495b-a085-06ebcbf58349"}, {"cell_type": "code", "source": "!wget http://hgdownload.soe.ucsc.edu/goldenPath/hg38/bigZips/hg38.fa.gz\n", "metadata": {"trusted": true, "tags": []}, "execution_count": 4, "outputs": [{"name": "stdout", "text": "--2025-01-25 23:33:06--  http://hgdownload.soe.ucsc.edu/goldenPath/hg38/bigZips/hg38.fa.gz\nResolving hgdownload.soe.ucsc.edu (hgdownload.soe.ucsc.edu)... 128.114.119.163\nConnecting to hgdownload.soe.ucsc.edu (hgdownload.soe.ucsc.edu)|128.114.119.163|:80... connected.\nHTTP request sent, awaiting response... 200 OK\nLength: 983659424 (938M) [application/x-gzip]\nSaving to: \u2018hg38.fa.gz\u2019\n\nhg38.fa.gz          100%[===================>] 938.09M  26.1MB/s    in 37s     \n\n2025-01-25 23:33:47 (25.0 MB/s) - \u2018hg38.fa.gz\u2019 saved [983659424/983659424]\n\n", "output_type": "stream"}], "id": "31db310a-6bff-43a0-a196-7fe746fbc97c"}, {"cell_type": "code", "source": "\n!gunzip hg38.fa.gz", "metadata": {"trusted": true, "tags": []}, "execution_count": 5, "outputs": [], "id": "df18e7f0-7831-40db-8989-2e89ed4ebb54"}, {"cell_type": "code", "source": "!dx upload hg38.fa", "metadata": {"trusted": true, "tags": []}, "execution_count": 6, "outputs": [{"name": "stdout", "text": "[===========================================================>] Uploaded 3,273,481,150 of 3,273,481,150 bytes (100%) hg38.fa\nID                                file-GyBgB9jJbP2V5fkpB7jPYZzf\nClass                             file\nProject                           project-GkYf2zQJbP2Q3vFgf14863Gf\nFolder                            /\nName                              hg38.fa\nState                             \u001b[33mclosing\u001b[0m\nVisibility                        visible\nTypes                             -\nProperties                        -\nTags                              -\nOutgoing links                    -\nCreated                           Sat Jan 25 23:35:03 2025\nCreated by                        vidhu\n via the job                      job-GyBf69jJbP2z23Xzppk3FZVV\nLast modified                     Sat Jan 25 23:35:09 2025\nMedia type                        \narchivalState                     \"live\"\ncloudAccount                      \"cloudaccount-dnanexus\"\n", "output_type": "stream"}], "id": "faffa277-c72f-445b-b12d-9fd49ea304a9"}, {"cell_type": "markdown", "source": "## Left-align and normalize", "metadata": {"tags": []}, "id": "2b8f43a9-7b0c-4622-8651-5329845ec66c"}, {"cell_type": "code", "source": "\"\"\"\n------------------------------------------------------------------------------------------------------\n------------------------------------------------------------------------------------------------------\n------------------------------------------------------------------------------------------------------\n------------------------------ PAUSE HERE UNTIL PREVIOUS STEP COMPLETES ------------------------------\n------------------------------------------------------------------------------------------------------\n------------------------------------------------------------------------------------------------------\n------------------------------------------------------------------------------------------------------\n\"\"\"", "metadata": {"collapsed": true, "jupyter": {"outputs_hidden": true}}, "execution_count": null, "outputs": [], "id": "a960869a-67e1-464d-8ce8-b5b8cb666411"}, {"cell_type": "code", "source": "%%bash\n\ndx run swiss-army-knife \\\n-iin=\"/03_normalized/biallelic.vcf\" \\\n-iin=\"hg38.fa\" \\\n-icmd=\"bcftools norm -f hg38.fa -o normalized.vcf biallelic.vcf\" \\\n--instance-type mem2_ssd1_v2_x64 \\\n--destination \"${projectid}:/03_normalized/\"\n", "metadata": {"trusted": true, "tags": []}, "execution_count": 7, "outputs": [{"name": "stdout", "text": "\nUsing input JSON:\n{\n    \"cmd\": \"bcftools norm -f hg38.fa -o normalized.vcf biallelic.vcf\",\n    \"in\": [\n        {\n            \"$dnanexus_link\": {\n                \"project\": \"project-GkYf2zQJbP2Q3vFgf14863Gf\",\n                \"id\": \"file-GyBbz6jJZB03yYgj68y0Q4fx\"\n            }\n        },\n        {\n            \"$dnanexus_link\": {\n                \"project\": \"project-GkYf2zQJbP2Q3vFgf14863Gf\",\n                \"id\": \"file-GyBgB9jJbP2V5fkpB7jPYZzf\"\n            }\n        }\n    ]\n}\n\nCalling app-Gx52kJQ9GVfXJK4Zpk9x4ZJf with output destination\n  project-GkYf2zQJbP2Q3vFgf14863Gf:/03_normalized\n\nJob ID: job-GyBgBXQJbP2x7x9j0pp7PyZp\n", "output_type": "stream"}], "id": "cf1c6e99-309b-4ae0-9590-fa8191aef782"}, {"cell_type": "markdown", "source": "# Burden analysis", "metadata": {"tags": []}, "id": "6cfdd019-d7cf-404c-b863-f0d0eda2cef0"}, {"cell_type": "code", "source": "! wget https://github.com/zhanxw/rvtests/releases/download/v2.1.0/rvtests_linux64.tar.gz\n! tar -xvzf  rvtests_linux64.tar.gz\n", "metadata": {"trusted": true, "tags": []}, "execution_count": 2, "outputs": [{"name": "stdout", "text": "--2025-05-27 16:58:35--  https://github.com/zhanxw/rvtests/releases/download/v2.1.0/rvtests_linux64.tar.gz\nResolving github.com (github.com)... 20.26.156.215\nConnecting to github.com (github.com)|20.26.156.215|:443... connected.\nHTTP request sent, awaiting response... 302 Found\nLocation: https://objects.githubusercontent.com/github-production-release-asset-2e65be/7528862/63fbb780-29ae-11e9-91d3-99ec3c3134cc?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=releaseassetproduction%2F20250527%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20250527T165835Z&X-Amz-Expires=300&X-Amz-Signature=c64a4e896a7ca29986824d8bc45d49caf8fa2293797324b4d4d3a3add8bbb3ad&X-Amz-SignedHeaders=host&response-content-disposition=attachment%3B%20filename%3Drvtests_linux64.tar.gz&response-content-type=application%2Foctet-stream [following]\n--2025-05-27 16:58:35--  https://objects.githubusercontent.com/github-production-release-asset-2e65be/7528862/63fbb780-29ae-11e9-91d3-99ec3c3134cc?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=releaseassetproduction%2F20250527%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20250527T165835Z&X-Amz-Expires=300&X-Amz-Signature=c64a4e896a7ca29986824d8bc45d49caf8fa2293797324b4d4d3a3add8bbb3ad&X-Amz-SignedHeaders=host&response-content-disposition=attachment%3B%20filename%3Drvtests_linux64.tar.gz&response-content-type=application%2Foctet-stream\nResolving objects.githubusercontent.com (objects.githubusercontent.com)... 185.199.108.133, 185.199.109.133, 185.199.110.133, ...\nConnecting to objects.githubusercontent.com (objects.githubusercontent.com)|185.199.108.133|:443... connected.\nHTTP request sent, awaiting response... 200 OK\nLength: 38050737 (36M) [application/octet-stream]\nSaving to: \u2018rvtests_linux64.tar.gz\u2019\n\nrvtests_linux64.tar 100%[===================>]  36.29M  69.4MB/s    in 0.5s    \n\n2025-05-27 16:58:36 (69.4 MB/s) - \u2018rvtests_linux64.tar.gz\u2019 saved [38050737/38050737]\n\nexecutable/\nexecutable/combineKinship\nexecutable/plink2vcf\nexecutable/vcf2plink\nexecutable/vcfAnnoSummaryLite\nexecutable/explainCSI2\nexecutable/vcf2kinship\nexecutable/vcfIndvSummary\nexecutable/createVCFIndex\nexecutable/extractVCFIndex\nexecutable/explainTabix\nexecutable/vcf2ld_neighbor\nexecutable/vcfConcordance\nexecutable/vcfExtractSite\nexecutable/vcfSummaryLite\nexecutable/bgenFileInfo\nexecutable/vcfPeek\nexecutable/vcfVariantSummaryLite\nexecutable/kinshipDecompose\nexecutable/rvtest\nexecutable/vcfPair\nexecutable/vcfSummary\nexecutable/explainCSI1\nexecutable/vcf2ld_window\nexecutable/queryVCFIndex\nexecutable/bgen2vcf\nexecutable/vcf2ld_gene\nexecutable/vcf2geno\nexample/\nexample/README.md\nexample/example.vcf\nexample/covar.missing\nexample/pheno\nexample/experimental/\nexample/experimental/1k.fam\nexample/experimental/1k.bed\nexample/experimental/1k.bim\nexample/experimental/1k.1m.bgen\nexample/experimental/cmd.sh\nexample/experimental/1k.1m.vcf.gz\nexample/experimental/1k.ped\nexample/experimental/1k.1m.samples\nexample/example.vcf.gz.tbi\nexample/example.vcf.gz\nexample/cmd.sh\nexample/setFile\nexample/covar\nREADME.md\n", "output_type": "stream"}], "id": "7b24f7d7-f6bb-4e1f-a9ae-e303acec38d1"}, {"cell_type": "code", "source": "! dx download /03_normalized/normalized.vcf --overwrite", "metadata": {"trusted": true, "tags": []}, "execution_count": 3, "outputs": [{"name": "stdout", "text": "\u001b[2K[===========================================================>] Completed 223,302,477,453 of 223,302,477,453 bytes (100%) /opt/notebooks/normalized.vcff\n", "output_type": "stream"}], "id": "ffd52fa2-4f32-40cc-b7e7-fba080dd3a2e"}, {"cell_type": "code", "source": "!grep ^'#' normalized.vcf", "metadata": {"trusted": true, "tags": []}, "execution_count": null, "outputs": [], "id": "5506fdb2-b0d2-41eb-bb6c-fd3edc39329d"}, {"cell_type": "code", "source": "! (grep ^'#' /03_normalized/normalized.vcf; grep -v ^'#' /03_normalized/normalized.vcf | sed 's:^chr::ig' | sort -k1,1n -k2,2n) > normalized_reformat.vcf", "metadata": {"trusted": true, "tags": []}, "execution_count": null, "outputs": [], "id": "8cd17070-a54b-42a8-a032-abd81da9dfec"}, {"cell_type": "code", "source": "! dx upload normalized_reformat.vcf --path /03_normalized/normalized_reformat.vcf", "metadata": {"trusted": true, "tags": []}, "execution_count": null, "outputs": [], "id": "3e4f53a7-9773-4f0d-ae45-75a838a10dd7"}, {"cell_type": "code", "source": "%%bash\n\ndx run swiss-army-knife \\\n-iin=\"/03_normalized/normalized_reformat.vcf\" \\\n-icmd=\"bgzip normalized_reformat.vcf -k\" \\\n--instance-type mem2_ssd1_v2_x32 \\\n--destination \"${projectid}:/03_normalized\"\n", "metadata": {"trusted": true, "tags": []}, "execution_count": null, "outputs": [], "id": "3834609d-9590-457c-be57-2d098556ba6c"}, {"cell_type": "code", "source": "%%bash\n\ndx run swiss-army-knife \\\n-iin=\"/03_normalized/normalized_reformat.vcf.gz\" \\\n-icmd=\"tabix -f -p vcf normalized_reformat.vcf.gz\" \\\n--instance-type mem2_ssd1_v2_x32 \\\n--destination \"${projectid}:/03_normalized\"\n", "metadata": {"trusted": true, "tags": []}, "execution_count": null, "outputs": [], "id": "520666f9-0b4e-4bd1-938e-8d0d7e204fac"}, {"cell_type": "code", "source": "! dx download /03_normalized/normalized_reformat.vcf.gz --overwrite\n! dx download /03_normalized/normalized_reformat.vcf.gz.tbi --overwrite\n", "metadata": {"trusted": true, "tags": []}, "execution_count": 4, "outputs": [{"name": "stdout", "text": "\u001b[2K[===========================================================>] Completed 25,463,831,167 of 25,463,831,167 bytes (100%) /opt/notebooks/normalized_reformat.vcf.gzz\ndxpy.utils.resolver.ResolutionError: Unable to resolve \"normalized_reformat.vcf.gz.tbi\" to a data object or folder name in '/03_normalized'\n", "output_type": "stream"}], "id": "18c8b326-183b-4c8f-9634-785538a1b202"}, {"cell_type": "code", "source": "! dx download Covar_HT_Dem_remove_PD_VD_FTD_Hu_CJD_Vs_HT_New.txt --overwrite\n! dx download Pheno_HT_Dem_remove_PD_VD_FTD_Hu_CJD_Vs_HT_New.txt --overwrite\n! dx download refFlat.txt --overwrite", "metadata": {"trusted": true, "tags": []}, "execution_count": 5, "outputs": [{"name": "stdout", "text": "\u001b[2K[===========================================================>] Completed 15,659,433 of 15,659,433 bytes (100%) /opt/notebooks/Covar_HT_Dem_remove_PD_VD_FTD_Hu_CJD_Vs_HT_New.txtt\n\u001b[2K[===========================================================>] Completed 1,634,665 of 1,634,665 bytes (100%) /opt/notebooks/Pheno_HT_Dem_remove_PD_VD_FTD_Hu_CJD_Vs_HT_New.txtt\n\u001b[2K[===========================================================>] Completed 22,201,141 of 22,201,141 bytes (100%) /opt/notebooks/refFlat.txtt\n", "output_type": "stream"}], "id": "19767c96-59f7-43ac-99ed-82ea6cc741b3"}, {"cell_type": "code", "source": "df_pheno = pd.read_csv(\"pheno.txt\", sep=\"\\t\", header=None)\ndf_pheno.rename({0:\"FID\",1:\"IID\",2:\"PHENO\"}, axis=1, inplace=True)\ndf_covar = pd.read_csv(\"covar.txt\", sep=\"\\t\")\ndf_covar_merged = df_covar.merge(df_pheno, on=[\"FID\",\"IID\"])\ndf_covar_merged.to_csv(\"covar_merged.txt\", sep=\"\\t\", index=False)\n", "metadata": {"trusted": true, "tags": []}, "execution_count": null, "outputs": [], "id": "028143d8-70ee-455a-bc7d-478b45704985"}, {"cell_type": "code", "source": "!gunzip -c normalized_reformat.vcf.gz > normalized_reformat.vcf", "metadata": {"trusted": true, "tags": []}, "execution_count": 14, "outputs": [], "id": "c58e3670-28d6-481e-9b4a-fefa750bc01a"}, {"cell_type": "code", "source": "!wc -l normalized_reformat.vcf", "metadata": {"trusted": true, "tags": []}, "execution_count": null, "outputs": [], "id": "b4b21bf9-b7e9-4473-830c-92c3a04b4b79"}, {"cell_type": "code", "source": "! executable/rvtest \\\n    --inVcf normalized_reformat.vcf \\\n    --out burden \\\n    --numThread 96 \\\n    --noweb \\\n    --hide-covar \\\n    --kernel skat,skato \\\n    --multipleAllele \\\n    --geneFile refFlat.txt \\\n    --gene CRIM1 \\\n    --pheno Covar_HT_Dem_remove_PD_VD_FTD_Hu_CJD_Vs_HT_New.txt \\\n    --pheno-name ADRD \\\n    --covar Covar_HT_Dem_remove_PD_VD_FTD_Hu_CJD_Vs_HT_New.txt \\\n    --covar-name GENETIC_SEX,AGE_2024_COV,PC1,PC2,PC3,PC4,PC5 \n", "metadata": {"trusted": true, "tags": []}, "execution_count": 15, "outputs": [{"name": "stdout", "text": "Thank you for using rvtests (version: 20190205, git: c86e589efef15382603300dc7f4c3394c82d69b8)\n  For documentations, refer to http://zhanxw.github.io/rvtests/\n  For questions and comments, plase send to Xiaowei Zhan <zhanxw@umich.edu>\n  For bugs and feature requests, please submit at: https://github.com/zhanxw/rvtests/issues\n\nThe following parameters are available.  Ones with \"[]\" are in effect:\n\nAvailable Options\n      Basic Input/Output: --inVcf [normalized_reformat.vcf], --inBgen []\n                          --inBgenSample [], --inKgg [], --out [burden]\n                          --outputRaw\n       Specify Covariate:\n                          --covar [Covar_HT_Dem_remove_PD_VD_FTD_Hu_CJD_Vs_HT_New.txt]\n                          --covar-name [GENETIC_SEX,AGE_2024_COV,PC1,PC2,PC3,PC4\n                         PC5], --sex\n       Specify Phenotype:\n                          --pheno [Covar_HT_Dem_remove_PD_VD_FTD_Hu_CJD_Vs_HT_New.txt]\n                          --inverseNormal, --useResidualAsPhenotype, --mpheno []\n                          --pheno-name [ADRD], --qtl, --multiplePheno []\n        Specify Genotype: --dosage [], --multipleAllele [true]\n    Chromosome X Options: --xLabel [], --xParRegion []\n           People Filter: --peopleIncludeID [], --peopleIncludeFile []\n                          --peopleExcludeID [], --peopleExcludeFile []\n             Site Filter: --rangeList [], --rangeFile [], --siteFile []\n                          --siteDepthMin [], --siteDepthMax [], --siteMACMin []\n                          --annoType []\n         Genotype Filter: --indvDepthMin [], --indvDepthMax [], --indvQualMin []\n       Association Model: --single [], --burden [], --vt [], --kernel [skat\n                         skato], --meta []\n     Family-based Models: --kinship [], --xHemiKinship [], --kinshipEigen []\n                          --xHemiKinshipEigen [], --boltPlink []\n                          --boltPlinkNoCheck\n          Grouping Unit : --geneFile [refFlat.txt], --gene [CRIM1], --setList []\n                          --setFile [], --set []\n        Frequency Cutoff: --freqUpper [], --freqLower []\n            Missing Data: --impute [], --imputePheno, --imputeCov\n    Conditional Analysis: --condition []\n     Auxiliary Functions: --noweb [true], --hide-covar [true], --numThread [96]\n                          --outputID, --help\n\nEffective Options\n    --inVcf normalized_reformat.vcf\n    --out burden\n    --covar Covar_HT_Dem_remove_PD_VD_FTD_Hu_CJD_Vs_HT_New.txt\n    --covar-name GENETIC_SEX,AGE_2024_COV,PC1,PC2,PC3,PC4,PC5\n    --pheno Covar_HT_Dem_remove_PD_VD_FTD_Hu_CJD_Vs_HT_New.txt\n    --pheno-name ADRD\n    --multipleAllele\n    --kernel skat,skato\n    --geneFile refFlat.txt\n    --gene CRIM1\n    --noweb\n    --hide-covar\n    --numThread 96\n\n[INFO]\tProgram version: 20190205\nReduced your specified number of threads to the maximum of system limit [ 16 ][INFO]\tAnalysis started at: Tue May 27 18:28:43 2025\nReport 'Wrong VCF header' to zhanxw@umich.edu\n", "output_type": "stream"}], "id": "be6093a3-88fa-428b-a6b7-f8f9d33b82fb"}, {"cell_type": "code", "source": "", "metadata": {}, "execution_count": null, "outputs": [], "id": "cfe7f6c1-0e71-48c8-9ffd-abc88aef93e2"}]}