{"metadata": {"kernelspec": {"display_name": "Python 3 (ipykernel)", "language": "python", "name": "python3"}, "language_info": {"name": "python", "version": "3.11.5", "mimetype": "text/x-python", "codemirror_mode": {"name": "ipython", "version": 3}, "pygments_lexer": "ipython3", "nbconvert_exporter": "python", "file_extension": ".py"}}, "nbformat_minor": 5, "nbformat": 4, "cells": [{"cell_type": "code", "source": "1.We extracted data from the \u201cUK Biobank\u201d, identifying 3,290 cases with essential hypertension and co-occurring dementia, after excluding individuals with Parkinson\u2019s disease, vascular dementia, frontotemporal dementia, Huntington\u2019s disease, or Creutzfeldt-Jakob disease. A total of 87,524 individuals with essential hypertension but without any neurodegenerative conditions were included as controls and ran genome-wide association analysis using PLINK.2.0. \n2.APOE status was calculated using https://github.com/neurogenetics/APOE_genotypes?tab=readme-ov-file#1.\n3.APOE status or APOE genotype was included in the covariate file.\n4.We ran GWAS using PLINK.2.0 and 10 principle components, sex_At_birth, current age (2024-year of birth), age filter > 60 and APOE status. ", "metadata": {"trusted": true, "tags": []}, "execution_count": 4, "outputs": [], "id": "5e6b79e5-621b-4738-b72e-084e1036aace"}, {"cell_type": "code", "source": "import pandas as pd\nimport subprocess\nimport sys\nimport numpy as np\nimport os\nimport shutil", "metadata": {}, "execution_count": null, "outputs": [], "id": "bcbe25bc-4161-448c-a023-94b5d996ac74"}, {"cell_type": "code", "source": "def shell_do(command, log=False, return_log=False):\n    print(f'Executing: {(\" \").join(command.split())}', file=sys.stderr)\n\n    res=subprocess.run(command.split(), stdout=subprocess.PIPE)\n\n    if log:\n        print(res.stdout.decode('utf-8'))\n    if return_log:\n        return(res.stdout.decode('utf-8'), res.stderr.decode('utf-8'))", "metadata": {"trusted": true, "tags": []}, "execution_count": 5, "outputs": [], "id": "1b2ae4ce-af25-4a5c-bb55-fb3bd84462c4"}, {"cell_type": "markdown", "source": "# Interacting with Swiss Army Knife through Jupyter Notebooks\n## What is Swiss Army Knife?\n### Swiss Army Knife is a generic app which can be used to perform common file operations for the genotype data housed on DNANexus.\n### Contains software such as plink, plink2, bcftools, etc.\n#### Below we'll see an example of looping through each chromosome and extracting a set of variants.\n##### -iin: input (need to specify each file).\n##### -icmd: command to run\n##### --instance-type: VM type to run the command on\n##### --destination: output folder\n#### Note: this will start 23 separate Swiss Army Knife jobs so make sure you are using a reasonable instance type.", "metadata": {"tags": []}, "id": "e0d68ce0-919f-4121-91f9-860e37a0cbeb"}, {"cell_type": "code", "source": "import pandas as pd", "metadata": {"trusted": true, "tags": []}, "execution_count": 6, "outputs": [], "id": "fb8fcc24-e79e-4337-931a-885a9a1bb301"}, {"cell_type": "code", "source": "!dx download ADRD_noFTD_noPARKINSONISM_noALS_noVD_noHT_noCJD_noPDD.CTRL_inclProxy_60older.plink_pheno.txt", "metadata": {"trusted": true, "tags": []}, "execution_count": 7, "outputs": [{"name": "stdout", "text": "Error: path \"/opt/notebooks/ADRD_noFTD_noPARKINSONISM_noALS_noVD_noHT_noCJD_no\nPDD.CTRL_inclProxy_60older.plink_pheno.txt\" already exists but -f/--overwrite\nwas not set\n", "output_type": "stream"}], "id": "2ba6d30b-f4fc-4037-b26a-4b8e6029073b"}, {"cell_type": "code", "source": "df = pd.read_csv(\"ADRD_noFTD_noPARKINSONISM_noALS_noVD_noHT_noCJD_noPDD.CTRL_inclProxy_60older.plink_pheno.txt\", sep = '\\t')", "metadata": {"trusted": true, "tags": []}, "execution_count": 8, "outputs": [], "id": "22615315-a34e-46fb-80b3-20571cc5566a"}, {"cell_type": "code", "source": "count_1 = df['ADRD'].value_counts().get(1, 0)\ncount_2 = df['ADRD'].value_counts().get(2, 0)\n\nprint(f\"Count of 1 in 'Pheno': {count_1}\")\nprint(f\"Count of 2 in 'Pheno': {count_2}\")", "metadata": {"trusted": true, "tags": []}, "execution_count": 9, "outputs": [{"name": "stdout", "text": "Count of 1 in 'Pheno': 87524\nCount of 2 in 'Pheno': 3290\n", "output_type": "stream"}], "id": "d105c0f6-51fe-4d16-a4ba-8eeb5af432f2"}, {"cell_type": "code", "source": "!dx download UKB_EUR_UMAP_COVARIATES.txt", "metadata": {"trusted": true, "tags": []}, "execution_count": 10, "outputs": [{"name": "stdout", "text": "Error: path \"/opt/notebooks/UKB_EUR_UMAP_COVARIATES.txt\" already exists but\n-f/--overwrite was not set\n", "output_type": "stream"}], "id": "17e7fa94-a0a5-4de6-b1b7-eb4b83f9700d"}, {"cell_type": "code", "source": "df1 = pd.read_csv(\"UKB_EUR_UMAP_COVARIATES.txt\", sep = '\\t')", "metadata": {"trusted": true, "tags": []}, "execution_count": 11, "outputs": [], "id": "e6c48734-c530-4f60-9f6e-69f0b83f3049"}, {"cell_type": "code", "source": "# Merging the DataFrames on the 'ID' column\nfinal = pd.merge(df, df1, on=['FID','IID'], how='inner')", "metadata": {"trusted": true, "tags": []}, "execution_count": 14, "outputs": [], "id": "4d592cbb-1c43-4704-bb55-80a3699e05fe"}, {"cell_type": "code", "source": "final.to_csv(\"Pheno_New_final_HT_Dem_remove_PD_VD_FTD_Hu_CJD_Vs_HT_New.txt\", sep='\\t', index=False)", "metadata": {"trusted": true, "tags": []}, "execution_count": 15, "outputs": [], "id": "113d20a0-6327-4ace-8c18-bc273899ed2e"}, {"cell_type": "code", "source": "!dx upload Pheno_New_final_HT_Dem_remove_PD_VD_FTD_Hu_CJD_Vs_HT_New.txt", "metadata": {"trusted": true, "tags": []}, "execution_count": 16, "outputs": [{"name": "stdout", "text": "[===========================================================>] Uploaded 15,659,433 of 15,659,433 bytes (100%) Pheno_New_final_HT_Dem_remove_PD_VD_FTD_Hu_CJD_Vs_HT_New.txt\nID                                file-J13XGq0JbP2qBpJB8yQPvpQp\nClass                             file\nProject                           project-GkYf2zQJbP2Q3vFgf14863Gf\nFolder                            /\nName                              Pheno_New_final_HT_Dem_remove_PD_VD_FTD_Hu_CJD_Vs_HT_New.txt\nState                             \u001b[33mclosing\u001b[0m\nVisibility                        visible\nTypes                             -\nProperties                        -\nTags                              -\nOutgoing links                    -\nCreated                           Mon Jun  9 18:22:36 2025\nCreated by                        vidhu\n via the job                      job-J13QB2jJbP2x0Pz725v7kPj7\nLast modified                     Mon Jun  9 18:22:37 2025\nMedia type                        \narchivalState                     \"live\"\ncloudAccount                      \"cloudaccount-dnanexus\"\n", "output_type": "stream"}], "id": "a557b2c0-f5f9-4e97-95c0-7c2eca004afa"}, {"cell_type": "code", "source": "%%bash\n\nseq 1 1 23 > chr_list.txt\n\nFILE_DIR_A=\"/Imputed_Genotypes_2023/TOPMed_Plink2\"\nFILE_DIR_B=\"/ref_panel/chip_overlaps/\"\n# FILE_DIR_C=\"/Imputed_Genotypes_2023/TOPMed_Plink2/extract\"\n\nCHR_IDS=$(cat chr_list.txt)\n\nfor CHR in $CHR_IDS; do\n    dx run swiss-army-knife -iin=\"chr${CHR}_pgen.pgen\" \\\n    -iin=\"chr${CHR}_pgen.psam\" \\\n    -iin=\"chr${CHR}_pgen.pvar\" \\\n    -icmd='plink2 --pfile \"$in_prefix\" \\\n    --snps 19:44908684:T:C,19:44908822:C:T --make-bed \\\n    --out \"$in_prefix\".apoe_snps' -y  --brief --priority normal \\\n    --instance-type mem3_ssd3_x12 --destination \"${projectid}:${FILE_DIR}/\" \ndone ", "metadata": {"trusted": true, "tags": []}, "execution_count": 16, "outputs": [{"name": "stdout", "text": "job-J0q4Kp8JbP2ZKZf9V5F37VY1\njob-J0q4KpQJbP2k5k7B5Z8kX34q\njob-J0q4Kq0JbP2pQPyYX2XPyZjb\njob-J0q4Kq8JbP2XJQ0Zvf2bpj18\njob-J0q4KqQJbP2jG3GGgxBzK7gB\njob-J0q4Kv0JbP2pQPyYX2XPyZjg\njob-J0q4Kv8JbP2pQPyYX2XPyZjk\njob-J0q4KvQJbP2pQPyYX2XPyZjq\njob-J0q4KvjJbP2Z4qz5fxzjkG6f\njob-J0q4Kx8JbP2XJQ0Zvf2bpj1F\njob-J0q4KxQJbP2V0VbPkP12vyPp\njob-J0q4KxjJbP2fBk9FQYQ20zvG\njob-J0q4Ky8JbP2V0VbPkP12vyPy\njob-J0q4KyQJbP2v55f99FF2K5Bk\njob-J0q4KyjJbP2vY2BvyyXPVz5Z\njob-J0q4Kz8JbP2V0VbPkP12vyQ4\njob-J0q4KzQJbP2yzY4kykqVKZ4b\njob-J0q4KzjJbP2fBk9FQYQ20zvf\njob-J0q4P00JbP2vY2BvyyXPVz60\njob-J0q4P0QJbP2V0VbPkP12vyQX\njob-J0q4P0jJbP2Z4qz5fxzjkG7Q\njob-J0q4P10JbP2V0VbPkP12vyQZ\njob-J0q4P1QJbP2vY2BvyyXPVz6K\n", "output_type": "stream"}], "id": "08d9831b-37b4-4d1b-a9d0-d09fb72d1fd3"}, {"cell_type": "code", "source": "%%bash\n\nCHR=19\n\nFILE_DIR_A=\"/Imputed_Genotypes_2023/TOPMed_Plink2\"\nFILE_DIR_B=\"/ref_panel/chip_overlaps/\"\n# FILE_DIR_C=\"/Imputed_Genotypes_2023/TOPMed_Plink2/extract\"\ndx run swiss-army-knife -iin=\"chr${CHR}_pgen.apoe_snps.bed\" \\\n    -iin=\"chr${CHR}_pgen.apoe_snps.bim\" \\\n    -iin=\"chr${CHR}_pgen.apoe_snps.fam\" \\\n    -icmd='plink --bfile \"$in_prefix\" \\\n    --allow-no-sex \\\n    --recodeA \\\n    --out \"$in_prefix\".apoe_snps_2' -y  --brief --priority normal \\\n    --instance-type mem3_ssd3_x12 --destination \"${projectid}:${FILE_DIR}/\"", "metadata": {}, "execution_count": null, "outputs": [], "id": "19cb8b8f-d6f1-45a2-b16c-3e39ac227f72"}, {"cell_type": "code", "source": "!dx download chr19_pgen.apoe_snps.apoe_snps.ped", "metadata": {"trusted": true, "tags": []}, "execution_count": 17, "outputs": [{"name": "stdout", "text": "\u001b[2K[===========================================================>] Completed 15,104,731 of 15,104,731 bytes (100%) /opt/notebooks/chr19_pgen.apoe_snps.apoe_snps.pedd\n", "output_type": "stream"}], "id": "5d1efdf8-aa4e-4646-b1cd-ec559875517a"}, {"cell_type": "code", "source": " # Import necessary packages\nimport numpy as np\nimport pandas as pd\nfrom functools import reduce\n\n# Define input and output file names directly in the notebook\ninput_file = 'chr19_pgen.apoe_snps.apoe_snps.ped'  # Replace with the actual path to your .ped file\noutput_name = 'chr19_pgen.apoe_snps.apoe_snps_out.ped'     # Replace with the desired output file name (without suffix)\n\n# Read in the .ped file and force column names\nheader_text = [\"FID\", \"IID\", \"PAT\", \"MAT\", \"SEX\", \"PHENO\", \"rs429358\", \"rs7412\"]\ninput_ped_df = pd.read_csv(input_file, sep='\\t', header=None, names=header_text)\n\n# Make a combined column by concatenating rs429358 and rs7412 genotypes\ninput_ped_df['rs429358_rs7412'] = input_ped_df['rs429358'].astype(str) + '_' + input_ped_df['rs7412']\n\n# Initialize a dictionary to map the combined genotypes to APOE genotypes\napoe_genotypes_dict = {\n    'CC_TT': 'e1/e1',\n    'CT_TT': 'e1/e2',\n    'TC_TT': 'e1/e2',\n    'CC_CT': 'e1/e4',\n    'CC_TC': 'e1/e4',\n    'TT_TT': 'e2/e2',\n    'TT_TC': 'e2/e3',\n    'TT_CT': 'e2/e3',\n    'TC_TC': 'e2/e4 or e1/e3',\n    'CT_CT': 'e2/e4 or e1/e3',\n    'TC_CT': 'e2/e4 or e1/e3',\n    'CT_TC': 'e2/e4 or e1/e3',\n    'TT_CC': 'e3/e3',\n    'TC_CC': 'e3/e4',\n    'CT_CC': 'e3/e4',\n    'CC_CC': 'e4/e4'\n}\n\n# Map the combined genotype column to APOE genotypes\ninput_ped_df['APOE_GENOTYPE'] = input_ped_df['rs429358_rs7412'].map(apoe_genotypes_dict)\n\n# Handle NaN values by marking unknown genotypes\ninput_ped_df['APOE_GENOTYPE'].fillna('unknown', inplace=True)\n\n# Create a new DataFrame with the relevant columns\nsubset_geno_df = input_ped_df[['FID', 'IID', 'SEX', 'PHENO', 'APOE_GENOTYPE']]\n\n## Generate counts and percentages\n\n# Overall APOE genotype counts and percentages\ncounts_df = pd.DataFrame(subset_geno_df['APOE_GENOTYPE'].value_counts().reset_index())\ncounts_df.columns = ['APOE_GENOTYPE', 'TOTAL_COUNT']\ncounts_df['TOTAL_PERCENT'] = (counts_df['TOTAL_COUNT'] / subset_geno_df.shape[0]) * 100\n\n# Separate into cases, controls, and missing phenotype\nmissing_pheno_df = subset_geno_df[subset_geno_df['PHENO'] == -9]\ncontrols_df = subset_geno_df[subset_geno_df['PHENO'] == 1]\ncases_df = subset_geno_df[subset_geno_df['PHENO'] == 2]\n\n# Generate counts and percentages for missing phenotypes\nmissing_pheno_counts_df = pd.DataFrame(missing_pheno_df['APOE_GENOTYPE'].value_counts().reset_index())\nmissing_pheno_counts_df.columns = ['APOE_GENOTYPE', 'MISSING_PHENO_COUNT']\nmissing_pheno_counts_df['MISSING_PHENO_PERCENT'] = (missing_pheno_counts_df['MISSING_PHENO_COUNT'] / missing_pheno_df.shape[0]) * 100\n\n# Generate counts and percentages for controls\ncontrols_counts_df = pd.DataFrame(controls_df['APOE_GENOTYPE'].value_counts().reset_index())\ncontrols_counts_df.columns = ['APOE_GENOTYPE', 'CONTROLS_COUNT']\ncontrols_counts_df['CONTROLS_PERCENT'] = (controls_counts_df['CONTROLS_COUNT'] / controls_df.shape[0]) * 100\n\n# Generate counts and percentages for cases\ncases_counts_df = pd.DataFrame(cases_df['APOE_GENOTYPE'].value_counts().reset_index())\ncases_counts_df.columns = ['APOE_GENOTYPE', 'CASES_COUNT']\ncases_counts_df['CASES_PERCENT'] = (cases_counts_df['CASES_COUNT'] / cases_df.shape[0]) * 100\n\n# Merge the dataframes to get a summary\ndataframes_tomerge = [counts_df, missing_pheno_counts_df, controls_counts_df, cases_counts_df]\nmerged_summary_df = reduce(lambda left, right: pd.merge(left, right, on='APOE_GENOTYPE', how='outer'), dataframes_tomerge)\n\n## Export results\ncomplete_df_output = output_name + \".APOE_GENOTYPES.csv\"\ncounts_df_output = output_name + \".APOE_SUMMARY.csv\"\n\n# Save the complete genotype file as .csv\nsubset_geno_df.to_csv(complete_df_output, index=False)\nprint(f\"Your complete genotype file has been saved here: {complete_df_output}\")\n\n# Save the summary counts file as .csv\nmerged_summary_df.to_csv(counts_df_output, index=False)\nprint(f\"The summary counts have been saved here: {counts_df_output}\")\n\n# Done!\nprint(\"Script execution complete!\")\n", "metadata": {"trusted": true, "tags": []}, "execution_count": 18, "outputs": [{"name": "stderr", "text": "/tmp/ipykernel_410/483976865.py:41: FutureWarning: A value is trying to be set on a copy of a DataFrame or Series through chained assignment using an inplace method.\nThe behavior will change in pandas 3.0. This inplace method will never work because the intermediate object on which we are setting values always behaves as a copy.\n\nFor example, when doing 'df[col].method(value, inplace=True)', try using 'df.method({col: value}, inplace=True)' or df[col] = df[col].method(value) instead, to perform the operation inplace on the original object.\n\n\n  input_ped_df['APOE_GENOTYPE'].fillna('unknown', inplace=True)\n", "output_type": "stream"}, {"name": "stdout", "text": "Your complete genotype file has been saved here: chr19_pgen.apoe_snps.apoe_snps_out.ped.APOE_GENOTYPES.csv\nThe summary counts have been saved here: chr19_pgen.apoe_snps.apoe_snps_out.ped.APOE_SUMMARY.csv\nScript execution complete!\n", "output_type": "stream"}], "id": "73ef7fce-248f-47a2-916a-31b891b54e7b"}, {"cell_type": "code", "source": "subset_geno_df['APOE_GENOTYPE'] = subset_geno_df['APOE_GENOTYPE'].replace({\n    'e4/e4': 2,\n    'e3/e4': 1\n}).map({2: 2, 1: 1}).fillna(0).astype(int)\n", "metadata": {"trusted": true, "tags": []}, "execution_count": 19, "outputs": [{"name": "stderr", "text": "/tmp/ipykernel_410/3959952741.py:1: SettingWithCopyWarning: \nA value is trying to be set on a copy of a slice from a DataFrame.\nTry using .loc[row_indexer,col_indexer] = value instead\n\nSee the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n  subset_geno_df['APOE_GENOTYPE'] = subset_geno_df['APOE_GENOTYPE'].replace({\n", "output_type": "stream"}], "id": "173a8af7-0fbb-4123-bf33-558bba8f94ea"}, {"cell_type": "code", "source": "subset_geno_df['APOE_GENOTYPE'] = subset_geno_df['APOE_GENOTYPE'].replace({\n    'e4/e4': 2,\n    'e3/e4': 1\n}).map({2: 2, 1: 1}).fillna(0).astype(int)\n\n# Display the updated dataframe\nsubset_geno_df.to_csv(\"APOE_Status.txt\", sep='\\t', index=False)", "metadata": {"trusted": true, "tags": []}, "execution_count": 20, "outputs": [{"name": "stderr", "text": "/tmp/ipykernel_410/749067298.py:1: SettingWithCopyWarning: \nA value is trying to be set on a copy of a slice from a DataFrame.\nTry using .loc[row_indexer,col_indexer] = value instead\n\nSee the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n  subset_geno_df['APOE_GENOTYPE'] = subset_geno_df['APOE_GENOTYPE'].replace({\n", "output_type": "stream"}], "id": "574ec6c7-e8f3-49a0-9394-50e98082bf01"}, {"cell_type": "code", "source": "!dx upload APOE_Status.txt", "metadata": {"trusted": true, "tags": []}, "execution_count": 21, "outputs": [{"name": "stdout", "text": "[===========================================================>] Uploaded 11,206,531 of 11,206,531 bytes (100%) APOE_Status.txt\nID                                file-J13XGy8JbP2qBpJB8yQPvpV8\nClass                             file\nProject                           project-GkYf2zQJbP2Q3vFgf14863Gf\nFolder                            /\nName                              APOE_Status.txt\nState                             \u001b[33mclosing\u001b[0m\nVisibility                        visible\nTypes                             -\nProperties                        -\nTags                              -\nOutgoing links                    -\nCreated                           Mon Jun  9 18:22:49 2025\nCreated by                        vidhu\n via the job                      job-J13QB2jJbP2x0Pz725v7kPj7\nLast modified                     Mon Jun  9 18:22:50 2025\nMedia type                        \narchivalState                     \"live\"\ncloudAccount                      \"cloudaccount-dnanexus\"\n", "output_type": "stream"}], "id": "372a59ae-74ac-4260-8e00-c19bd9811ae3"}, {"cell_type": "code", "source": "!dx upload chr19_pgen.apoe_snps.apoe_snps_out.ped.APOE_GENOTYPES.csv", "metadata": {"trusted": true, "tags": []}, "execution_count": 22, "outputs": [{"name": "stdout", "text": "[===========================================================>] Uploaded 13,265,397 of 13,265,397 bytes (100%) chr19_pgen.apoe_snps.apoe_snps_out.ped.APOE_GENOTYPES.csv\nID                                file-J13XGz0JbP2ZvK40VyKVyzx3\nClass                             file\nProject                           project-GkYf2zQJbP2Q3vFgf14863Gf\nFolder                            /\nName                              chr19_pgen.apoe_snps.apoe_snps_out.ped.APOE_GENOTYPES.csv\nState                             \u001b[33mclosing\u001b[0m\nVisibility                        visible\nTypes                             -\nProperties                        -\nTags                              -\nOutgoing links                    -\nCreated                           Mon Jun  9 18:22:52 2025\nCreated by                        vidhu\n via the job                      job-J13QB2jJbP2x0Pz725v7kPj7\nLast modified                     Mon Jun  9 18:22:53 2025\nMedia type                        \narchivalState                     \"live\"\ncloudAccount                      \"cloudaccount-dnanexus\"\n", "output_type": "stream"}], "id": "eb7a1044-f687-4450-8bde-e42f453394d4"}, {"cell_type": "code", "source": "!dx upload chr19_pgen.apoe_snps.apoe_snps_out.ped.APOE_SUMMARY.csv", "metadata": {"trusted": true, "tags": []}, "execution_count": 23, "outputs": [{"name": "stdout", "text": "[===========================================================>] Uploaded 569 of 569 bytes (100%) chr19_pgen.apoe_snps.apoe_snps_out.ped.APOE_SUMMARY.csv\nID                                file-J13XGzQJbP2ZvK40VyKVyzx5\nClass                             file\nProject                           project-GkYf2zQJbP2Q3vFgf14863Gf\nFolder                            /\nName                              chr19_pgen.apoe_snps.apoe_snps_out.ped.APOE_SUMMARY.csv\nState                             \u001b[33mclosing\u001b[0m\nVisibility                        visible\nTypes                             -\nProperties                        -\nTags                              -\nOutgoing links                    -\nCreated                           Mon Jun  9 18:22:54 2025\nCreated by                        vidhu\n via the job                      job-J13QB2jJbP2x0Pz725v7kPj7\nLast modified                     Mon Jun  9 18:22:55 2025\nMedia type                        \narchivalState                     \"live\"\ncloudAccount                      \"cloudaccount-dnanexus\"\n", "output_type": "stream"}], "id": "66a63739-2170-4552-a6a6-44a0e5355d56"}, {"cell_type": "code", "source": "!head chr19_pgen.apoe_snps.apoe_snps_out.ped.APOE_SUMMARY.csv", "metadata": {"trusted": true, "tags": []}, "execution_count": 24, "outputs": [{"name": "stdout", "text": "APOE_GENOTYPE,TOTAL_COUNT,TOTAL_PERCENT,MISSING_PHENO_COUNT,MISSING_PHENO_PERCENT,CONTROLS_COUNT,CONTROLS_PERCENT,CASES_COUNT,CASES_PERCENT\ne2/e2,3125,0.6413163711138793,3125,0.6413163711138793,,,,\ne2/e3,59312,12.172082113122052,59312,12.172082113122052,,,,\ne2/e4 or e1/e3,12132,2.4897440685931467,12132,2.4897440685931467,,,,\ne3/e3,286304,58.755661540924194,286304,58.755661540924194,,,,\ne3/e4,114542,23.50645112964031,114542,23.50645112964031,,,,\ne4/e4,11583,2.3770776085158607,11583,2.3770776085158607,,,,\nunknown,281,0.05766716809056003,281,0.05766716809056003,,,,\n", "output_type": "stream"}], "id": "94643617-2f9f-4241-b44d-27984ff8370a"}, {"cell_type": "code", "source": "!dx download ADRD_noFTD_noPARKINSONISM_noALS_noVD_noHT_noCJD_noPDD.CTRL_inclProxy_60older.plink_pheno.txt", "metadata": {"trusted": true, "tags": []}, "execution_count": 25, "outputs": [{"name": "stdout", "text": "Error: path \"/opt/notebooks/ADRD_noFTD_noPARKINSONISM_noALS_noVD_noHT_noCJD_no\nPDD.CTRL_inclProxy_60older.plink_pheno.txt\" already exists but -f/--overwrite\nwas not set\n", "output_type": "stream"}], "id": "cfc56cfd-2c75-452a-bef3-1e60b2814020"}, {"cell_type": "code", "source": "import pandas as pd\n\n# Read the two files\nfile1 = pd.read_csv('APOE_Status.txt', sep='\\t')\nfile2 = pd.read_csv('ADRD_noFTD_noPARKINSONISM_noALS_noVD_noHT_noCJD_noPDD.CTRL_inclProxy_60older.plink_pheno.txt', sep='\\t')\n\n# Merge the files on the common 'FID' and 'IID' columns\nmerged_df = pd.merge(file1, file2, on=['FID', 'IID'], how='inner')\n\nmerged_df.to_csv(\"HT_Alzheimers_APOE_Status.txt\", sep='\\t', index=False)\n", "metadata": {"trusted": true, "tags": []}, "execution_count": 26, "outputs": [], "id": "a26913cc-d3af-41aa-bc79-f843b18b3a4a"}, {"cell_type": "code", "source": "!dx download UKB_EUR_UMAP_COVARIATES.txt", "metadata": {"trusted": true, "tags": []}, "execution_count": 27, "outputs": [{"name": "stdout", "text": "Error: path \"/opt/notebooks/UKB_EUR_UMAP_COVARIATES.txt\" already exists but\n-f/--overwrite was not set\n", "output_type": "stream"}], "id": "0f847b2e-ec8b-40be-979c-c0465943a083"}, {"cell_type": "code", "source": "import pandas as pd\n# Read the two files\nfile11 = pd.read_csv('HT_Alzheimers_APOE_Status.txt', sep='\\t')\nfile21 = pd.read_csv('UKB_EUR_UMAP_COVARIATES.txt', sep='\\t')\n\n# Merge the files on the common 'FID' and 'IID' columns\nmerged_df1 = pd.merge(file11, file21, on=['FID', 'IID'], how='inner')\n\nmerged_df1.to_csv(\"HT_Alzheimers_APOE_Status_with_PCs.txt\", sep='\\t', index=False)", "metadata": {"trusted": true, "tags": []}, "execution_count": 28, "outputs": [], "id": "58ea682f-d2f4-4f1a-872a-516c6569915d"}, {"cell_type": "code", "source": "# Step 2: Specify the columns you want to extract\ncolumns_to_extract = ['FID', 'IID', 'APOE_GENOTYPE','PC1','PC2','PC3','PC4','PC5','PC6','PC7','PC8','PC9','PC10','GENETIC_SEX','AGE_2024_COV']  # Replace with actual column names\n\n# Step 3: Extract the specified columns\ndf_subset1 = merged_df1[columns_to_extract]\n\n# Step 4: Save the extracted columns to a new file if needed\ndf_subset1.to_csv('Covar_HT_Alzheimers_APOE_Status_with_PCs.txt', sep='\\t', index=False)", "metadata": {"trusted": true, "tags": []}, "execution_count": 29, "outputs": [], "id": "9adafeb4-0766-45a9-8d50-d27d5ee6a651"}, {"cell_type": "code", "source": "# Step 2: Specify the columns you want to extract\ncolumns_to_extract = ['FID', 'IID', 'ADRD']  # Replace with actual column names\n\n# Step 3: Extract the specified columns\ndf_subset = merged_df1[columns_to_extract]\n\n# Step 4: Save the extracted columns to a new file if needed\ndf_subset.to_csv('Pheno_HT_Alzheimers_APOE_Status_with_PCs.txt', sep='\\t', index=False)", "metadata": {"trusted": true, "tags": []}, "execution_count": 30, "outputs": [], "id": "935e8e0c-388d-43fa-af65-bfd0ba4d8b5e"}, {"cell_type": "code", "source": "!dx upload Pheno_HT_Alzheimers_APOE_Status_with_PCs.txt", "metadata": {"trusted": true, "tags": []}, "execution_count": 31, "outputs": [{"name": "stdout", "text": "[===========================================================>] Uploaded 1,634,665 of 1,634,665 bytes (100%) Pheno_HT_Alzheimers_APOE_Status_with_PCs.txt\nID                                file-J13XJ1jJbP2pJGJf4B7QJb3K\nClass                             file\nProject                           project-GkYf2zQJbP2Q3vFgf14863Gf\nFolder                            /\nName                              Pheno_HT_Alzheimers_APOE_Status_with_PCs.txt\nState                             \u001b[33mclosing\u001b[0m\nVisibility                        visible\nTypes                             -\nProperties                        -\nTags                              -\nOutgoing links                    -\nCreated                           Mon Jun  9 18:23:03 2025\nCreated by                        vidhu\n via the job                      job-J13QB2jJbP2x0Pz725v7kPj7\nLast modified                     Mon Jun  9 18:23:04 2025\nMedia type                        \narchivalState                     \"live\"\ncloudAccount                      \"cloudaccount-dnanexus\"\n", "output_type": "stream"}], "id": "4d80182a-11ad-41f5-be87-932c3fe0fa6e"}, {"cell_type": "code", "source": "!dx upload Covar_HT_Alzheimers_APOE_Status_with_PCs.txt", "metadata": {"trusted": true, "tags": []}, "execution_count": 32, "outputs": [{"name": "stdout", "text": "[===========================================================>] Uploaded 13,481,565 of 13,481,565 bytes (100%) Covar_HT_Alzheimers_APOE_Status_with_PCs.txt\nID                                file-J13XJ2QJbP2pJGJf4B7QJb3X\nClass                             file\nProject                           project-GkYf2zQJbP2Q3vFgf14863Gf\nFolder                            /\nName                              Covar_HT_Alzheimers_APOE_Status_with_PCs.txt\nState                             \u001b[33mclosing\u001b[0m\nVisibility                        visible\nTypes                             -\nProperties                        -\nTags                              -\nOutgoing links                    -\nCreated                           Mon Jun  9 18:23:06 2025\nCreated by                        vidhu\n via the job                      job-J13QB2jJbP2x0Pz725v7kPj7\nLast modified                     Mon Jun  9 18:23:06 2025\nMedia type                        \narchivalState                     \"live\"\ncloudAccount                      \"cloudaccount-dnanexus\"\n", "output_type": "stream"}], "id": "5cacf75f-9eef-4cf3-90e8-3a9be90a5d53"}, {"cell_type": "code", "source": "!dx upload HT_Alzheimers_APOE_Status_with_PCs.txt", "metadata": {"trusted": true, "tags": []}, "execution_count": 34, "outputs": [{"name": "stdout", "text": "[===========================================================>] Uploaded 16,295,155 of 16,295,155 bytes (100%) HT_Alzheimers_APOE_Status_with_PCs.txt\nID                                file-J13XK2QJbP2VkVF7YY9fZjYy\nClass                             file\nProject                           project-GkYf2zQJbP2Q3vFgf14863Gf\nFolder                            /\nName                              HT_Alzheimers_APOE_Status_with_PCs.txt\nState                             \u001b[33mclosing\u001b[0m\nVisibility                        visible\nTypes                             -\nProperties                        -\nTags                              -\nOutgoing links                    -\nCreated                           Mon Jun  9 18:25:14 2025\nCreated by                        vidhu\n via the job                      job-J13QB2jJbP2x0Pz725v7kPj7\nLast modified                     Mon Jun  9 18:25:14 2025\nMedia type                        \narchivalState                     \"live\"\ncloudAccount                      \"cloudaccount-dnanexus\"\n", "output_type": "stream"}], "id": "cb9ef9b3-499e-4728-a762-665c3e77e8a6"}, {"cell_type": "code", "source": "# Use this if the file is CSV\ndf = pd.read_csv('HT_Alzheimers_APOE_Status_with_PCs.txt', sep='\\t', on_bad_lines='skip')", "metadata": {"trusted": true, "tags": []}, "execution_count": 43, "outputs": [], "id": "ff49d79b-bf33-436a-a095-5593e317d7ed"}, {"cell_type": "code", "source": "df.to_csv(\"HT_Alzheimers_APOE_Status_with_PCs_1.txt\", sep='\\t', index=False)", "metadata": {"trusted": true, "tags": []}, "execution_count": 44, "outputs": [], "id": "ce2c7c20-cb32-43d4-9177-108371fbd9bb"}, {"cell_type": "code", "source": "# Count unique values for each column\ncount_a = df['ADRD'].value_counts()\ncount_b = df['ADRD'].value_counts()\n\nprint(count_a)\nprint(count_b)", "metadata": {"trusted": true, "tags": []}, "execution_count": 46, "outputs": [{"name": "stdout", "text": "ADRD\n1    87524\n2     3290\nName: count, dtype: int64\nADRD\n1    87524\n2     3290\nName: count, dtype: int64\n", "output_type": "stream"}], "id": "4cdc610f-3013-4245-96a3-36652dfcb472"}, {"cell_type": "code", "source": "!dx upload HT_Alzheimers_APOE_Status_with_PCs_1.txt", "metadata": {"trusted": true, "tags": []}, "execution_count": 47, "outputs": [{"name": "stdout", "text": "[===========================================================>] Uploaded 16,295,155 of 16,295,155 bytes (100%) HT_Alzheimers_APOE_Status_with_PCs_1.txt\nID                                file-J13XPZ8JbP2pqfq6B3f9Fjj6\nClass                             file\nProject                           project-GkYf2zQJbP2Q3vFgf14863Gf\nFolder                            /\nName                              HT_Alzheimers_APOE_Status_with_PCs_1.txt\nState                             \u001b[33mclosing\u001b[0m\nVisibility                        visible\nTypes                             -\nProperties                        -\nTags                              -\nOutgoing links                    -\nCreated                           Mon Jun  9 18:28:33 2025\nCreated by                        vidhu\n via the job                      job-J13QB2jJbP2x0Pz725v7kPj7\nLast modified                     Mon Jun  9 18:28:34 2025\nMedia type                        \narchivalState                     \"live\"\ncloudAccount                      \"cloudaccount-dnanexus\"\n", "output_type": "stream"}], "id": "c5ae7300-1fc0-458e-8982-5ee62e2758c4"}, {"cell_type": "code", "source": "!dx upload HT_Alzheimers_APOE_Status_with_PCs.txt", "metadata": {"trusted": true, "tags": []}, "execution_count": 48, "outputs": [{"name": "stdout", "text": "[===========================================================>] Uploaded 16,295,155 of 16,295,155 bytes (100%) HT_Alzheimers_APOE_Status_with_PCs.txt\nID                                file-J13XPZjJbP2Q2kP0fb5GY44Y\nClass                             file\nProject                           project-GkYf2zQJbP2Q3vFgf14863Gf\nFolder                            /\nName                              HT_Alzheimers_APOE_Status_with_PCs.txt\nState                             \u001b[33mclosing\u001b[0m\nVisibility                        visible\nTypes                             -\nProperties                        -\nTags                              -\nOutgoing links                    -\nCreated                           Mon Jun  9 18:28:35 2025\nCreated by                        vidhu\n via the job                      job-J13QB2jJbP2x0Pz725v7kPj7\nLast modified                     Mon Jun  9 18:28:36 2025\nMedia type                        \narchivalState                     \"live\"\ncloudAccount                      \"cloudaccount-dnanexus\"\n", "output_type": "stream"}], "id": "3fd64c04-bf56-435a-a54f-ddfa2d897fd2"}, {"cell_type": "code", "source": "import pandas as pd\n\n# Assuming the dataframe is named 'df'\n# Example: df = pd.read_csv('HT_Alzheimers_APOE_Status_with_PCs.txt', delim_whitespace=True)\n\n# Extract the FID and IID columns\nfid_iid_df = merged_df1[['FID', 'IID']]\nfid_iid_df.to_csv(\"FID_IID_HT_Alzheimers_APOE_Status_with_PCs.txt\", sep='\\t', index=False)", "metadata": {"trusted": true, "tags": []}, "execution_count": 50, "outputs": [], "id": "25c65d03-dbff-4634-abab-8f9870f95a65"}, {"cell_type": "code", "source": "!dx upload FID_IID_HT_Alzheimers_APOE_Status_with_PCs.txt", "metadata": {"trusted": true, "tags": []}, "execution_count": 51, "outputs": [{"name": "stdout", "text": "[===========================================================>] Uploaded 1,453,032 of 1,453,032 bytes (100%) FID_IID_HT_Alzheimers_APOE_Status_with_PCs.txt\nID                                file-J13XPg8JbP2Q2kP0fb5GY44z\nClass                             file\nProject                           project-GkYf2zQJbP2Q3vFgf14863Gf\nFolder                            /\nName                              FID_IID_HT_Alzheimers_APOE_Status_with_PCs.txt\nState                             \u001b[33mclosing\u001b[0m\nVisibility                        visible\nTypes                             -\nProperties                        -\nTags                              -\nOutgoing links                    -\nCreated                           Mon Jun  9 18:28:45 2025\nCreated by                        vidhu\n via the job                      job-J13QB2jJbP2x0Pz725v7kPj7\nLast modified                     Mon Jun  9 18:28:46 2025\nMedia type                        \narchivalState                     \"live\"\ncloudAccount                      \"cloudaccount-dnanexus\"\n", "output_type": "stream"}], "id": "0be575fd-cc4c-4f28-90f8-65eba07e13cb"}, {"cell_type": "code", "source": "%%bash\n\nseq 1 1 23 > chr_list.txt\n\nFILE_DIR_A=\"/Imputed_Genotypes_2023/TOPMed_Plink2\"\nFILE_DIR_B=\"/ref_panel/chip_overlaps/\"\n# FILE_DIR_C=\"/Imputed_Genotypes_2023/TOPMed_Plink2/extract\"\n\nCHR_IDS=$(cat chr_list.txt)\n\n\nfor CHR in $CHR_IDS; do\n    dx run swiss-army-knife -iin=\"chr${CHR}_pgen.pgen\" \\\n    -iin=\"chr${CHR}_pgen.psam\" \\\n    -iin=\"chr${CHR}_pgen.pvar\" \\\n    -iin=\"Pheno_HT_Alzheimers_APOE_Status_with_PCs.txt\" \\\n    -iin=\"Covar_HT_Alzheimers_APOE_Status_with_PCs.txt\" \\\n    -iin=\"FID_IID_HT_Alzheimers_APOE_Status_with_PCs.txt\" \\\n    -icmd='plink2 --pfile \"$in_prefix\" \\\n    --glm hide-covar firth-fallback cols=+a1freq,+a1freqcc,+a1countcc,+totallelecc,+err,+beta,-test,-nobs,-tz,-orbeta \\\n    --pheno Pheno_HT_Alzheimers_APOE_Status_with_PCs.txt --maf 0.05 \\\n    --pheno-name ADRD --covar-variance-standardize \\\n    --geno 0.05 --mac 20  --memory 12000 \\\n    --covar Covar_HT_Alzheimers_APOE_Status_with_PCs.txt \\\n    --covar-name PC1,PC2,PC3,PC4,PC5,PC6,PC7,PC8,PC9,PC10,GENETIC_SEX,AGE_2024_COV,APOE_GENOTYPE \\\n    --keep FID_IID_HT_Alzheimers_APOE_Status_with_PCs.txt \\\n    --out \"$in_prefix\".Pheno_New_final_HT_with_APOE_Status_New' -y  --brief --priority normal \\\n    --instance-type mem3_ssd3_x12 --destination \"${projectid}:${FILE_DIR}/\" \ndone ", "metadata": {}, "execution_count": null, "outputs": [], "id": "9d887090-0050-46b0-b286-08257b8a8649"}]}